{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TODAY'S DATE\n",
      "Thu Mar 14 07:43:39 PDT 2019\n",
      "------------\n",
      "\n",
      "Distributor ID:\tUbuntu\n",
      "Description:\tUbuntu 16.04.5 LTS\n",
      "Release:\t16.04\n",
      "Codename:\txenial\n",
      "\n",
      "------------\n",
      "HOSTNAME: \n",
      "jpg-d20\n",
      "\n",
      "------------\n",
      "Computer Specs:\n",
      "\n",
      "Architecture:          x86_64\n",
      "CPU op-mode(s):        32-bit, 64-bit\n",
      "Byte Order:            Little Endian\n",
      "CPU(s):                24\n",
      "On-line CPU(s) list:   0-23\n",
      "Thread(s) per core:    2\n",
      "Core(s) per socket:    6\n",
      "Socket(s):             2\n",
      "NUMA node(s):          2\n",
      "Vendor ID:             GenuineIntel\n",
      "CPU family:            6\n",
      "Model:                 44\n",
      "Model name:            Intel(R) Xeon(R) CPU           X5680  @ 3.33GHz\n",
      "Stepping:              2\n",
      "CPU MHz:               2394.000\n",
      "CPU max MHz:           3326.0000\n",
      "CPU min MHz:           1596.0000\n",
      "BogoMIPS:              6666.90\n",
      "Virtualization:        VT-x\n",
      "L1d cache:             32K\n",
      "L1i cache:             32K\n",
      "L2 cache:              256K\n",
      "L3 cache:              12288K\n",
      "NUMA node0 CPU(s):     0,2,4,6,8,10,12,14,16,18,20,22\n",
      "NUMA node1 CPU(s):     1,3,5,7,9,11,13,15,17,19,21,23\n",
      "Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 popcnt aes lahf_lm epb kaiser tpr_shadow vnmi flexpriority ept vpid dtherm ida arat\n",
      "\n",
      "------------\n",
      "\n",
      "Memory Specs\n",
      "\n",
      "              total        used        free      shared  buff/cache   available\n",
      "Mem:            94G        2.2G        2.7G        1.1G         89G         90G\n",
      "Swap:          128G          0B        128G\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No LSB modules are available.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"TODAY'S DATE\"\n",
    "date\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "lsb_release -a\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"HOSTNAME: \"\n",
    "hostname\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"Computer Specs:\"\n",
    "echo \"\"\n",
    "lscpu\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "echo \"Memory Specs\"\n",
    "echo \"\"\n",
    "free -mh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: salmon_out_dir=/media/sam/4TB_toshiba/porites/20190314_salmon_abundance_porites\n",
      "env: coral_transcriptome=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta\n",
      "env: coral_fasta_index=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.fai\n",
      "env: samples=/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_male_vs_female_k5_trinity_sample_list.txt\n",
      "env: gene_map=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.gene_trans_map\n",
      "env: salmon_stdout=20190314_salmon.out\n",
      "env: salmon_stderr=20190314_salmon.err\n"
     ]
    }
   ],
   "source": [
    "%env salmon_out_dir=/media/sam/4TB_toshiba/porites/20190314_salmon_abundance_porites\n",
    "%env coral_transcriptome=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta\n",
    "%env coral_fasta_index=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.fai\n",
    "%env samples=/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_male_vs_female_k5_trinity_sample_list.txt\n",
    "%env gene_map=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.gene_trans_map\n",
    "\n",
    "%env salmon_stdout=20190314_salmon.out\n",
    "%env salmon_stderr=20190314_salmon.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "male  male_bleached_K5_01  4R041-L7-P01-AGTTCC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P01-AGTTCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "male  male_bleached_K5_02  4R041-L7-P06-GCCAAT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P06-GCCAAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "male  male_bleached_K5_03  4R041-L7-P07-AGTCAA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P07-AGTCAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "female  female_bleached_K5_04  4R041-L7-P08-CTTGTA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P08-CTTGTA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "female  female_bleached_K5_05  4R041-L7-P09-GTGAAA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P09-GTGAAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "female  female_bleached_K5_06  4R041-L7-P13-ATCACG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P13-ATCACG-READ2-Sequences.txt.gz_val_2.fq.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat ${samples}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "real\t5m30.372s\n",
      "user\t105m24.856s\n",
      "sys\t0m57.288s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /mnt/data/coral_RNAseq_2017/porites/20180311_fastqc_trimming/trimmed\n",
    "\n",
    "time  /home/shared/Trinityrnaseq-v2.6.6/util/align_and_estimate_abundance.pl \\\n",
    "--output_dir ${salmon_out_dir} \\\n",
    "--transcripts ${coral_transcriptome} \\\n",
    "--seqType fq \\\n",
    "--samples_file ${samples} \\\n",
    "--SS_lib_type RF \\\n",
    "--est_method salmon \\\n",
    "--aln_method bowtie2 \\\n",
    "--trinity_mode \\\n",
    "--prep_reference \\\n",
    "--thread_count 23 \\\n",
    "1> ${salmon_out_dir}/${salmon_stdout} \\\n",
    "2> ${salmon_out_dir}/${salmon_stderr}\n",
    "sed '/^Subject:/ s/ / porites_abundance JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 34G\n",
      "drwxrwxr-x 2 sam sam  4.0K Apr 15  2018 fastqc\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P01-AGTTCC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P01-AGTTCC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 15  2018 4R041-L7-P01-AGTTCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P01-AGTTCC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 15  2018 4R041-L7-P02-ACAGTG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  998M Apr 15  2018 4R041-L7-P02-ACAGTG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 15  2018 4R041-L7-P02-ACAGTG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P02-ACAGTG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P03-ACTGAT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  956M Apr 15  2018 4R041-L7-P03-ACTGAT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P03-ACTGAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P03-ACTGAT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 15  2018 4R041-L7-P04-GTCCGC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  941M Apr 15  2018 4R041-L7-P04-GTCCGC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P04-GTCCGC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P04-GTCCGC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 15  2018 4R041-L7-P05-GAGTGG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P05-GAGTGG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 15  2018 4R041-L7-P05-GAGTGG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.9K Apr 15  2018 4R041-L7-P05-GAGTGG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P06-GCCAAT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  761M Apr 15  2018 4R041-L7-P06-GCCAAT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  859M Apr 15  2018 4R041-L7-P06-GCCAAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 15  2018 4R041-L7-P06-GCCAAT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P07-AGTCAA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 16  2018 4R041-L7-P07-AGTCAA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 16  2018 4R041-L7-P07-AGTCAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P07-AGTCAA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P08-CTTGTA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 16  2018 4R041-L7-P08-CTTGTA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.4G Apr 16  2018 4R041-L7-P08-CTTGTA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P08-CTTGTA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 16  2018 4R041-L7-P09-GTGAAA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 1022M Apr 16  2018 4R041-L7-P09-GTGAAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  955M Apr 16  2018 4R041-L7-P09-GTGAAA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P09-GTGAAA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P10-TAGCTT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.4G Apr 16  2018 4R041-L7-P10-TAGCTT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.6G Apr 16  2018 4R041-L7-P10-TAGCTT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P10-TAGCTT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P11-ACTTGA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  916M Apr 16  2018 4R041-L7-P11-ACTTGA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 16  2018 4R041-L7-P11-ACTTGA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 16  2018 4R041-L7-P11-ACTTGA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P12-TGACCA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 16  2018 4R041-L7-P12-TGACCA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.3G Apr 16  2018 4R041-L7-P12-TGACCA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P12-TGACCA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P13-ATCACG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  760M Apr 16  2018 4R041-L7-P13-ATCACG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  786M Apr 16  2018 4R041-L7-P13-ATCACG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 16  2018 4R041-L7-P13-ATCACG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P14-GTGGCC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.4G Apr 16  2018 4R041-L7-P14-GTGGCC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.5G Apr 16  2018 4R041-L7-P14-GTGGCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P14-GTGGCC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 16  2018 4R041-L7-P15-GTTTCG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.5G Apr 16  2018 4R041-L7-P15-GTTTCG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.6G Apr 16  2018 4R041-L7-P15-GTTTCG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P15-GTTTCG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  220K Apr 16  2018 stderr.log\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar 14 07:47 male_bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar 14 07:48 male_bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar 14 07:49 male_bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar 14 07:50 female_bleached_K5_04\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar 14 07:51 female_bleached_K5_05\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar 14 07:51 female_bleached_K5_06\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -ltrh /mnt/data/coral_RNAseq_2017/porites/20180311_fastqc_trimming/trimmed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 84K\n",
      "drwxrwxr-x 2 sam sam 4.0K Apr 15  2018 fastqc\n",
      "-rw-rw-r-- 1 sam sam    0 Mar 14 07:46 20190314_salmon.out\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:47 male_bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:48 male_bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:49 male_bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:50 female_bleached_K5_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:51 female_bleached_K5_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:51 female_bleached_K5_06\n",
      "-rw-rw-r-- 1 sam sam  52K Mar 14 07:51 20190314_salmon.err\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mv /mnt/data/coral_RNAseq_2017/porites/20180311_fastqc_trimming/trimmed/[mf]* \\\n",
    "${salmon_out_dir}\n",
    "\n",
    "ls -ltrh ${salmon_out_dir}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 2.0M\n",
      "drwxrwxr-x 2 sam sam 4.0K Apr 15  2018 fastqc\n",
      "-rw-rw-r-- 1 sam sam    0 Mar 14 07:46 20190314_salmon.out\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:47 male_bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:48 male_bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:49 male_bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:50 female_bleached_K5_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:51 female_bleached_K5_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar 14 07:51 female_bleached_K5_06\n",
      "-rw-rw-r-- 1 sam sam  52K Mar 14 07:51 20190314_salmon.err\n",
      "-rw-rw-r-- 1 sam sam 425K Mar 14 09:05 salmon.isoform.TPM.not_cross_norm\n",
      "-rw-rw-r-- 1 sam sam 492K Mar 14 09:05 salmon.isoform.counts.matrix\n",
      "-rw-rw-r-- 1 sam sam 235K Mar 14 09:05 salmon.gene.TPM.not_cross_norm\n",
      "-rw-rw-r-- 1 sam sam 177K Mar 14 09:05 salmon.gene.counts.matrix\n",
      "-rw-rw-r-- 1 sam sam  528 Mar 14 09:05 salmon.isoform.TPM.not_cross_norm.runTMM.R\n",
      "-rw-rw-r-- 1 sam sam  418 Mar 14 09:05 salmon.isoform.TPM.not_cross_norm.TMM_info.txt\n",
      "-rw-rw-r-- 1 sam sam 381K Mar 14 09:05 salmon.isoform.TMM.EXPR.matrix\n",
      "-rw-rw-r-- 1 sam sam  522 Mar 14 09:05 salmon.gene.TPM.not_cross_norm.runTMM.R\n",
      "-rw-rw-r-- 1 sam sam  421 Mar 14 09:05 salmon.gene.TPM.not_cross_norm.TMM_info.txt\n",
      "-rw-rw-r-- 1 sam sam 189K Mar 14 09:05 salmon.gene.TMM.EXPR.matrix\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-reading file: male_bleached_K5_03/quant.sf\n",
      "-reading file: female_bleached_K5_06/quant.sf\n",
      "-reading file: female_bleached_K5_04/quant.sf\n",
      "-reading file: male_bleached_K5_02/quant.sf\n",
      "-reading file: male_bleached_K5_01/quant.sf\n",
      "-reading file: female_bleached_K5_05/quant.sf\n",
      "\n",
      "\n",
      "* Outputting combined matrix.\n",
      "\n",
      "/home/shared/Trinityrnaseq-v2.6.6/util/support_scripts/run_TMM_scale_matrix.pl --matrix salmon.isoform.TPM.not_cross_norm > salmon.isoform.TMM.EXPR.matrixCMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.TPM.not_cross_norm.runTMM.R 1>&2 \n",
      "> library(edgeR)\n",
      "Loading required package: limma\n",
      "> \n",
      "> rnaseqMatrix = read.table(\"salmon.isoform.TPM.not_cross_norm\", header=T, row.names=1, com='', check.names=F)\n",
      "> rnaseqMatrix = as.matrix(rnaseqMatrix)\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=factor(colnames(rnaseqMatrix)))\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study$samples$eff.lib.size = exp_study$samples$lib.size * exp_study$samples$norm.factors\n",
      "> write.table(exp_study$samples, file=\"salmon.isoform.TPM.not_cross_norm.TMM_info.txt\", quote=F, sep=\"\\t\", row.names=F)\n",
      "> \n",
      "/home/shared/Trinityrnaseq-v2.6.6/util/support_scripts/run_TMM_scale_matrix.pl --matrix salmon.gene.TPM.not_cross_norm > salmon.gene.TMM.EXPR.matrixCMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.gene.TPM.not_cross_norm.runTMM.R 1>&2 \n",
      "> library(edgeR)\n",
      "Loading required package: limma\n",
      "> \n",
      "> rnaseqMatrix = read.table(\"salmon.gene.TPM.not_cross_norm\", header=T, row.names=1, com='', check.names=F)\n",
      "> rnaseqMatrix = as.matrix(rnaseqMatrix)\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=factor(colnames(rnaseqMatrix)))\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study$samples$eff.lib.size = exp_study$samples$lib.size * exp_study$samples$norm.factors\n",
      "> write.table(exp_study$samples, file=\"salmon.gene.TPM.not_cross_norm.TMM_info.txt\", quote=F, sep=\"\\t\", row.names=F)\n",
      "> \n",
      "\n",
      "real\t0m2.037s\n",
      "user\t0m1.956s\n",
      "sys\t0m0.060s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ${salmon_out_dir}\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/util/abundance_estimates_to_matrix.pl \\\n",
    "--est_method salmon \\\n",
    "--gene_trans_map ${gene_map} \\\n",
    "--out_prefix salmon \\\n",
    "--name_sample_by_basedir \\\n",
    "male_bleached_K5_03/quant.sf \\\n",
    "female_bleached_K5_06/quant.sf \\\n",
    "female_bleached_K5_04/quant.sf \\\n",
    "male_bleached_K5_02/quant.sf \\\n",
    "male_bleached_K5_01/quant.sf \\\n",
    "female_bleached_K5_05/quant.sf\n",
    "\n",
    "ls -ltrh\n",
    "\n",
    "sed '/^Subject:/ s/ / porites_abundances JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "male_bleached_K5_03        female_bleached_K5_06  female_bleached_K5_04  male_bleached_K5_02  male_bleached_K5_01  female_bleached_K5_05\n",
      "TRINITY_DN181253_c0_g1_i6  31.000000              95.000000              0.000000             3.000000             0.000000               79.000000\n",
      "TRINITY_DN186288_c0_g2_i2  198.007926             0.000000               62.394878            0.000000             285.075425             35.260819\n",
      "TRINITY_DN193323_c0_g1_i7  18.271255              0.000000               0.000000             8.473519             40.062348              46.476249\n",
      "TRINITY_DN145218_c0_g1_i1  0.001004               0.000557               0.000000             0.000000             0.001022               0.000000\n",
      "TRINITY_DN197048_c2_g1_i1  0.000000               5.360445               0.000000             0.000000             0.000000               0.000000\n",
      "TRINITY_DN150956_c0_g1_i1  60.437768              8.000000               35.000000            19.465379            54.711420              34.748115\n",
      "TRINITY_DN186148_c2_g1_i8  3.086805               1.606200               7.079828             21.281485            0.000000               5.453134\n",
      "TRINITY_DN196742_c2_g1_i3  4839.000000            3619.000000            7780.000000          4712.000000          8595.000000            4933.000000\n",
      "TRINITY_DN174781_c0_g1_i2  1516.282550            2.985409               840.225894           1075.119853          2861.029648            17.300052\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "head ${salmon_out_dir}/salmon.isoform.counts.matrix \\\n",
    "| column -t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "$VAR1 = {\n",
      "          'male' => [\n",
      "                      'male_bleached_K5_01',\n",
      "                      'male_bleached_K5_02',\n",
      "                      'male_bleached_K5_03'\n",
      "                    ],\n",
      "          'female' => [\n",
      "                        'female_bleached_K5_04',\n",
      "                        'female_bleached_K5_05',\n",
      "                        'female_bleached_K5_06'\n",
      "                      ]\n",
      "        };\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.female_vs_male.female.vs.male.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/porites/20190314_salmon_abundance_porites/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(3,6,2,5,4,1)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"female\", 3), rep(\"male\", 3)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"female\", \"male\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"female\", sampleB=\"male\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.female_vs_male.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.female_vs_male.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.female_vs_male.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Got 6 samples, and got: 7 data fields.\n",
      "Header: male_bleached_K5_03\tfemale_bleached_K5_06\tfemale_bleached_K5_04\tmale_bleached_K5_02\tmale_bleached_K5_01\tfemale_bleached_K5_05\n",
      "Next: TRINITY_DN181253_c0_g1_i6\t31.000000\t95.000000\t0.000000\t3.000000\t0.000000\t79.000000\n",
      "\n",
      "$VAR1 = {\n",
      "          'male_bleached_K5_02' => 4,\n",
      "          'female_bleached_K5_05' => 6,\n",
      "          'male_bleached_K5_03' => 1,\n",
      "          'male_bleached_K5_01' => 5,\n",
      "          'female_bleached_K5_06' => 2,\n",
      "          'female_bleached_K5_04' => 3\n",
      "        };\n",
      "Contrasts to perform are: $VAR1 = [\n",
      "          [\n",
      "            'female',\n",
      "            'male'\n",
      "          ]\n",
      "        ];\n",
      "Loading required package: limma\n",
      "\n",
      "real\t0m2.395s\n",
      "user\t0m2.324s\n",
      "sys\t0m0.040s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/run_DE_analysis.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.counts.matrix \\\n",
    "--method edgeR \\\n",
    "--samples_file ${samples} \n",
    "\n",
    "sed '/^Subject:/ s/ / porites_edgeR JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 14M\n",
      "-rw-rw-r-- 1 sam sam  13M Jan 30 11:09 Trinity.sp.coral.fasta\n",
      "-rw-rw-r-- 1 sam sam 257K Jan 30 11:09 Trinity.sp.coral.fasta.fai\n",
      "-rw-rw-r-- 1 sam sam 280K Jan 30 21:09 Trinity.sp.coral.fasta.gene_trans_map\n",
      "drwxrwxr-x 2 sam sam 4.0K Jan 30 21:09 Trinity.sp.coral.fasta.salmon_quasi.idx\n",
      "drwxrwxr-x 2 sam sam 4.0K Jan 31 08:00 edgeR.3209.dir\n",
      "drwxrwxr-x 2 sam sam  12K Mar  6 08:58 edgeR.24054.dir\n",
      "drwxrwxr-x 2 sam sam 4.0K Mar 14 09:06 edgeR.13936.dir\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -ltrh /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> library(cluster)\n",
      "> library(Biobase)\n",
      "> library(qvalue)\n",
      "> library(fastcluster)\n",
      "> options(stringsAsFactors = FALSE)\n",
      "> NO_REUSE = F\n",
      "> \n",
      "> # try to reuse earlier-loaded data if possible\n",
      "> if (file.exists(\"diffExpr.P0.001_C2.matrix.RData\") && ! NO_REUSE) {\n",
      "+     print('RESTORING DATA FROM EARLIER ANALYSIS')\n",
      "+     load(\"diffExpr.P0.001_C2.matrix.RData\")\n",
      "+ } else {\n",
      "+     print('Reading matrix file.')\n",
      "+     primary_data = read.table(\"diffExpr.P0.001_C2.matrix\", header=T, com='', row.names=1, check.names=F)\n",
      "+     primary_data = as.matrix(primary_data)\n",
      "+ }\n",
      "[1] \"Reading matrix file.\"\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/heatmap.3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/misc_rnaseq_funcs.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/pairs3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/vioplot2.R\")\n",
      "> data = primary_data\n",
      "> myheatcol = colorpanel(75, 'purple','black','yellow')\n",
      "> samples_data = read.table(\"/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_male_vs_female_k5_trinity_sample_list.txt\", header=F, check.names=F, fill=T)\n",
      "> samples_data = samples_data[samples_data[,2] != '',]\n",
      "> colnames(samples_data) = c('sample_name', 'replicate_name')\n",
      "> sample_types = as.character(unique(samples_data[,1]))\n",
      "> rep_names = as.character(samples_data[,2])\n",
      "> data = data[, colnames(data) %in% rep_names, drop=F ]\n",
      "> nsamples = length(sample_types)\n",
      "> sample_colors = rainbow(nsamples)\n",
      "> names(sample_colors) = sample_types\n",
      "> sample_type_list = list()\n",
      "> for (i in 1:nsamples) {\n",
      "+     samples_want = samples_data[samples_data[,1]==sample_types[i], 2]\n",
      "+     sample_type_list[[sample_types[i]]] = as.vector(samples_want)\n",
      "+ }\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> initial_matrix = data # store before doing various data transformations\n",
      "> data = log2(data+1)\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> sampleAnnotations = matrix(ncol=ncol(data),nrow=nsamples)\n",
      "> for (i in 1:nsamples) {\n",
      "+   sampleAnnotations[i,] = colnames(data) %in% sample_type_list[[sample_types[i]]]\n",
      "+ }\n",
      "> sampleAnnotations = apply(sampleAnnotations, 1:2, function(x) as.logical(x))\n",
      "> sampleAnnotations = sample_matrix_to_color_assignments(sampleAnnotations, col=sample_colors)\n",
      "> rownames(sampleAnnotations) = as.vector(sample_types)\n",
      "> colnames(sampleAnnotations) = colnames(data)\n",
      "> data = as.matrix(data) # convert to matrix\n",
      "> \n",
      "> # Centering rows\n",
      "> data = t(scale(t(data), scale=F))\n",
      "> \n",
      "> write.table(data, file=\"diffExpr.P0.001_C2.matrix.log2.centered.dat\", quote=F, sep='\t');\n",
      "> if (nrow(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two rows are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> if (ncol(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two columns are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> sample_cor = cor(data, method='pearson', use='pairwise.complete.obs')\n",
      "> write.table(sample_cor, file=\"diffExpr.P0.001_C2.matrix.log2.centered.sample_cor.dat\", quote=F, sep='\t')\n",
      "> sample_dist = dist(t(data), method='euclidean')\n",
      "> hc_samples = hclust(sample_dist, method='complete')\n",
      "> pdf(\"diffExpr.P0.001_C2.matrix.log2.centered.sample_cor_matrix.pdf\")\n",
      "> sample_cor_for_plot = sample_cor\n",
      "> heatmap.3(sample_cor_for_plot, dendrogram='both', Rowv=as.dendrogram(hc_samples), Colv=as.dendrogram(hc_samples), col = myheatcol, scale='none', symm=TRUE, key=TRUE,density.info='none', trace='none', symkey=FALSE, symbreaks=F, margins=c(10,10), cexCol=1, cexRow=1, cex.main=0.75, main=paste(\"sample correlation matrix\n",
      "+ \", \"diffExpr.P0.001_C2.matrix.log2.centered\") , ColSideColors=sampleAnnotations, RowSideColors=t(sampleAnnotations))\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> gene_cor = NULL\n",
      "> gene_dist = dist(data, method='euclidean')\n",
      "> if (nrow(data) <= 1) { message('Too few genes to generate heatmap'); quit(status=0); }\n",
      "> hc_genes = hclust(gene_dist, method='complete')\n",
      "> heatmap_data = data\n",
      "> pdf(\"diffExpr.P0.001_C2.matrix.log2.centered.genes_vs_samples_heatmap.pdf\")\n",
      "> heatmap.3(heatmap_data, dendrogram='both', Rowv=as.dendrogram(hc_genes), Colv=as.dendrogram(hc_samples), col=myheatcol, scale=\"none\", density.info=\"none\", trace=\"none\", key=TRUE, keysize=1.2, cexCol=1, margins=c(10,10), cex.main=0.75, main=paste(\"samples vs. features\n",
      "+ \", \"diffExpr.P0.001_C2.matrix.log2.centered\" ) , ColSideColors=sampleAnnotations)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> save(list=ls(all=TRUE), file=\"diffExpr.P0.001_C2.matrix.RData\")\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "** Found 8 features as differentially expressed.\n",
      "\n",
      "CMD: /home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/PtR -m diffExpr.P0.001_C2.matrix --log2 --heatmap --min_colSums 0 --min_rowSums 0 --gene_dist euclidean --sample_dist euclidean --sample_cor_matrix --center_rows --save  -s /home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_male_vs_female_k5_trinity_sample_list.txt\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < diffExpr.P0.001_C2.matrix.R\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘fastcluster’\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    hclust\n",
      "\n",
      "for plotting:: min.raw: -0.990555804908208 max.raw: 1\n",
      "for plotting:: min.raw: -6.23274792537913 max.raw: 6.23274792537913\n",
      "\n",
      "real\t0m1.699s\n",
      "user\t0m1.624s\n",
      "sys\t0m0.060s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.13936.dir/\n",
    "\n",
    "time /home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/analyze_diff_expr.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.TMM.EXPR.matrix \\\n",
    "--samples ${samples}\n",
    "\n",
    "sed '/^Subject:/ s/ / porites_edgeR JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> library(cluster)\n",
      "> library(Biobase)\n",
      "> library(qvalue)\n",
      "> library(fastcluster)\n",
      "> options(stringsAsFactors = FALSE)\n",
      "> NO_REUSE = F\n",
      "> \n",
      "> # try to reuse earlier-loaded data if possible\n",
      "> if (file.exists(\"diffExpr.P0.05_C1.matrix.RData\") && ! NO_REUSE) {\n",
      "+     print('RESTORING DATA FROM EARLIER ANALYSIS')\n",
      "+     load(\"diffExpr.P0.05_C1.matrix.RData\")\n",
      "+ } else {\n",
      "+     print('Reading matrix file.')\n",
      "+     primary_data = read.table(\"diffExpr.P0.05_C1.matrix\", header=T, com='', row.names=1, check.names=F)\n",
      "+     primary_data = as.matrix(primary_data)\n",
      "+ }\n",
      "[1] \"Reading matrix file.\"\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/heatmap.3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/misc_rnaseq_funcs.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/pairs3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/vioplot2.R\")\n",
      "> data = primary_data\n",
      "> myheatcol = colorpanel(75, 'purple','black','yellow')\n",
      "> samples_data = read.table(\"/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_male_vs_female_k5_trinity_sample_list.txt\", header=F, check.names=F, fill=T)\n",
      "> samples_data = samples_data[samples_data[,2] != '',]\n",
      "> colnames(samples_data) = c('sample_name', 'replicate_name')\n",
      "> sample_types = as.character(unique(samples_data[,1]))\n",
      "> rep_names = as.character(samples_data[,2])\n",
      "> data = data[, colnames(data) %in% rep_names, drop=F ]\n",
      "> nsamples = length(sample_types)\n",
      "> sample_colors = rainbow(nsamples)\n",
      "> names(sample_colors) = sample_types\n",
      "> sample_type_list = list()\n",
      "> for (i in 1:nsamples) {\n",
      "+     samples_want = samples_data[samples_data[,1]==sample_types[i], 2]\n",
      "+     sample_type_list[[sample_types[i]]] = as.vector(samples_want)\n",
      "+ }\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> initial_matrix = data # store before doing various data transformations\n",
      "> data = log2(data+1)\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> sampleAnnotations = matrix(ncol=ncol(data),nrow=nsamples)\n",
      "> for (i in 1:nsamples) {\n",
      "+   sampleAnnotations[i,] = colnames(data) %in% sample_type_list[[sample_types[i]]]\n",
      "+ }\n",
      "> sampleAnnotations = apply(sampleAnnotations, 1:2, function(x) as.logical(x))\n",
      "> sampleAnnotations = sample_matrix_to_color_assignments(sampleAnnotations, col=sample_colors)\n",
      "> rownames(sampleAnnotations) = as.vector(sample_types)\n",
      "> colnames(sampleAnnotations) = colnames(data)\n",
      "> data = as.matrix(data) # convert to matrix\n",
      "> \n",
      "> # Centering rows\n",
      "> data = t(scale(t(data), scale=F))\n",
      "> \n",
      "> write.table(data, file=\"diffExpr.P0.05_C1.matrix.log2.centered.dat\", quote=F, sep='\t');\n",
      "> if (nrow(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two rows are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> if (ncol(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two columns are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> sample_cor = cor(data, method='pearson', use='pairwise.complete.obs')\n",
      "> write.table(sample_cor, file=\"diffExpr.P0.05_C1.matrix.log2.centered.sample_cor.dat\", quote=F, sep='\t')\n",
      "> sample_dist = dist(t(data), method='euclidean')\n",
      "> hc_samples = hclust(sample_dist, method='complete')\n",
      "> pdf(\"diffExpr.P0.05_C1.matrix.log2.centered.sample_cor_matrix.pdf\")\n",
      "> sample_cor_for_plot = sample_cor\n",
      "> heatmap.3(sample_cor_for_plot, dendrogram='both', Rowv=as.dendrogram(hc_samples), Colv=as.dendrogram(hc_samples), col = myheatcol, scale='none', symm=TRUE, key=TRUE,density.info='none', trace='none', symkey=FALSE, symbreaks=F, margins=c(10,10), cexCol=1, cexRow=1, cex.main=0.75, main=paste(\"sample correlation matrix\n",
      "+ \", \"diffExpr.P0.05_C1.matrix.log2.centered\") , ColSideColors=sampleAnnotations, RowSideColors=t(sampleAnnotations))\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> gene_cor = NULL\n",
      "> gene_dist = dist(data, method='euclidean')\n",
      "> if (nrow(data) <= 1) { message('Too few genes to generate heatmap'); quit(status=0); }\n",
      "> hc_genes = hclust(gene_dist, method='complete')\n",
      "> heatmap_data = data\n",
      "> pdf(\"diffExpr.P0.05_C1.matrix.log2.centered.genes_vs_samples_heatmap.pdf\")\n",
      "> heatmap.3(heatmap_data, dendrogram='both', Rowv=as.dendrogram(hc_genes), Colv=as.dendrogram(hc_samples), col=myheatcol, scale=\"none\", density.info=\"none\", trace=\"none\", key=TRUE, keysize=1.2, cexCol=1, margins=c(10,10), cex.main=0.75, main=paste(\"samples vs. features\n",
      "+ \", \"diffExpr.P0.05_C1.matrix.log2.centered\" ) , ColSideColors=sampleAnnotations)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> save(list=ls(all=TRUE), file=\"diffExpr.P0.05_C1.matrix.RData\")\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "** Found 55 features as differentially expressed.\n",
      "\n",
      "CMD: /home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/PtR -m diffExpr.P0.05_C1.matrix --log2 --heatmap --min_colSums 0 --min_rowSums 0 --gene_dist euclidean --sample_dist euclidean --sample_cor_matrix --center_rows --save  -s /home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_male_vs_female_k5_trinity_sample_list.txt\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < diffExpr.P0.05_C1.matrix.R\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘fastcluster’\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    hclust\n",
      "\n",
      "for plotting:: min.raw: -0.793853369632211 max.raw: 1\n",
      "for plotting:: min.raw: -9.49941488027207 max.raw: 9.49941488027207\n",
      "\n",
      "real\t0m1.769s\n",
      "user\t0m1.680s\n",
      "sys\t0m0.076s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.13936.dir/\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/analyze_diff_expr.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.TMM.EXPR.matrix \\\n",
    "--samples ${samples} \\\n",
    "-C 1 \\\n",
    "-P 0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
