{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TODAY'S DATE\n",
      "Wed Jan 30 20:30:37 PST 2019\n",
      "------------\n",
      "\n",
      "Distributor ID:\tUbuntu\n",
      "Description:\tUbuntu 16.04.5 LTS\n",
      "Release:\t16.04\n",
      "Codename:\txenial\n",
      "\n",
      "------------\n",
      "HOSTNAME: \n",
      "jpg-d20\n",
      "\n",
      "------------\n",
      "Computer Specs:\n",
      "\n",
      "Architecture:          x86_64\n",
      "CPU op-mode(s):        32-bit, 64-bit\n",
      "Byte Order:            Little Endian\n",
      "CPU(s):                24\n",
      "On-line CPU(s) list:   0-23\n",
      "Thread(s) per core:    2\n",
      "Core(s) per socket:    6\n",
      "Socket(s):             2\n",
      "NUMA node(s):          2\n",
      "Vendor ID:             GenuineIntel\n",
      "CPU family:            6\n",
      "Model:                 44\n",
      "Model name:            Intel(R) Xeon(R) CPU           X5680  @ 3.33GHz\n",
      "Stepping:              2\n",
      "CPU MHz:               1596.000\n",
      "CPU max MHz:           3326.0000\n",
      "CPU min MHz:           1596.0000\n",
      "BogoMIPS:              6666.90\n",
      "Virtualization:        VT-x\n",
      "L1d cache:             32K\n",
      "L1i cache:             32K\n",
      "L2 cache:              256K\n",
      "L3 cache:              12288K\n",
      "NUMA node0 CPU(s):     0,2,4,6,8,10,12,14,16,18,20,22\n",
      "NUMA node1 CPU(s):     1,3,5,7,9,11,13,15,17,19,21,23\n",
      "Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 popcnt aes lahf_lm epb kaiser tpr_shadow vnmi flexpriority ept vpid dtherm ida arat\n",
      "\n",
      "------------\n",
      "\n",
      "Memory Specs\n",
      "\n",
      "              total        used        free      shared  buff/cache   available\n",
      "Mem:            94G        2.0G         45G        787M         47G         91G\n",
      "Swap:          128G          0B        128G\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No LSB modules are available.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"TODAY'S DATE\"\n",
    "date\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "lsb_release -a\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"HOSTNAME: \"\n",
    "hostname\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"Computer Specs:\"\n",
    "echo \"\"\n",
    "lscpu\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "echo \"Memory Specs\"\n",
    "echo \"\"\n",
    "free -mh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: salmon_out_dir=/media/sam/4TB_toshiba/porites/20190130_salmon\n",
      "env: coral_transcriptome=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta\n",
      "env: coral_fasta_index=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.fai\n",
      "env: samples=/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_trinity_sample_list.txt\n",
      "env: gene_map=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.gene_trans_map\n",
      "env: salmon_stdout=20190130_salmon.out\n",
      "env: salmon_stderr=20190130_salmon.err\n"
     ]
    }
   ],
   "source": [
    "%env salmon_out_dir=/media/sam/4TB_toshiba/porites/20190130_salmon\n",
    "%env coral_transcriptome=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta\n",
    "%env coral_fasta_index=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.fai\n",
    "%env samples=/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_trinity_sample_list.txt\n",
    "%env gene_map=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.gene_trans_map\n",
    "\n",
    "%env salmon_stdout=20190130_salmon.out\n",
    "%env salmon_stderr=20190130_salmon.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bleached  bleached_K5_01  4R041-L7-P01-AGTTCC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P01-AGTTCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_K5_02  4R041-L7-P06-GCCAAT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P06-GCCAAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_K5_03  4R041-L7-P07-AGTCAA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P07-AGTCAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_K5_04  4R041-L7-P08-CTTGTA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P08-CTTGTA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_K5_05  4R041-L7-P09-GTGAAA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P09-GTGAAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_K5_06  4R041-L7-P13-ATCACG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P13-ATCACG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_44_01  4R041-L7-P03-ACTGAT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P03-ACTGAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_44_02  4R041-L7-P11-ACTTGA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P11-ACTTGA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_44_03  4R041-L7-P14-GTGGCC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P14-GTGGCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_K5_01  4R041-L7-P02-ACAGTG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P02-ACAGTG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_K5_02  4R041-L7-P10-TAGCTT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P10-TAGCTT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_K5_03  4R041-L7-P15-GTTTCG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P15-GTTTCG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_44_01  4R041-L7-P04-GTCCGC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P04-GTCCGC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_44_02  4R041-L7-P05-GAGTGG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P05-GAGTGG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_44_03  4R041-L7-P12-TGACCA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P12-TGACCA-READ2-Sequences.txt.gz_val_2.fq.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat ${samples}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 5: /media/sam/4TB_toshiba/porites/20190130_salmon/20190130_salmon.out: No such file or directory\n",
      "\n",
      "real\t0m0.001s\n",
      "user\t0m0.000s\n",
      "sys\t0m0.000s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /mnt/data/coral_RNAseq_2017/porites/20180311_fastqc_trimming/trimmed\n",
    "\n",
    "time  /home/shared/Trinityrnaseq-v2.6.6/util/align_and_estimate_abundance.pl \\\n",
    "--output_dir ${salmon_out_dir} \\\n",
    "--transcripts ${coral_transcriptome} \\\n",
    "--seqType fq \\\n",
    "--samples_file ${samples} \\\n",
    "--SS_lib_type RF \\\n",
    "--est_method salmon \\\n",
    "--aln_method bowtie2 \\\n",
    "--trinity_mode \\\n",
    "--prep_reference \\\n",
    "--thread_count 23 \\\n",
    "1> ${salmon_out_dir}/${salmon_stdout} \\\n",
    "2> ${salmon_out_dir}/${salmon_stderr}\n",
    "sed '/^Subject:/ s/ / porites_rsem JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir ${salmon_out_dir}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "real\t15m51.773s\n",
      "user\t295m12.568s\n",
      "sys\t3m27.776s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /mnt/data/coral_RNAseq_2017/porites/20180311_fastqc_trimming/trimmed\n",
    "\n",
    "time  /home/shared/Trinityrnaseq-v2.6.6/util/align_and_estimate_abundance.pl \\\n",
    "--output_dir ${salmon_out_dir} \\\n",
    "--transcripts ${coral_transcriptome} \\\n",
    "--seqType fq \\\n",
    "--samples_file ${samples} \\\n",
    "--SS_lib_type RF \\\n",
    "--est_method salmon \\\n",
    "--aln_method bowtie2 \\\n",
    "--trinity_mode \\\n",
    "--prep_reference \\\n",
    "--thread_count 23 \\\n",
    "1> ${salmon_out_dir}/${salmon_stdout} \\\n",
    "2> ${salmon_out_dir}/${salmon_stderr}\n",
    "sed '/^Subject:/ s/ / porites_rsem JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 34G\n",
      "drwxrwxr-x 2 sam sam  4.0K Apr 15  2018 fastqc\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P01-AGTTCC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P01-AGTTCC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 15  2018 4R041-L7-P01-AGTTCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P01-AGTTCC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 15  2018 4R041-L7-P02-ACAGTG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  998M Apr 15  2018 4R041-L7-P02-ACAGTG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 15  2018 4R041-L7-P02-ACAGTG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P02-ACAGTG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P03-ACTGAT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  956M Apr 15  2018 4R041-L7-P03-ACTGAT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P03-ACTGAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P03-ACTGAT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 15  2018 4R041-L7-P04-GTCCGC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  941M Apr 15  2018 4R041-L7-P04-GTCCGC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P04-GTCCGC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P04-GTCCGC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 15  2018 4R041-L7-P05-GAGTGG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P05-GAGTGG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 15  2018 4R041-L7-P05-GAGTGG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.9K Apr 15  2018 4R041-L7-P05-GAGTGG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P06-GCCAAT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  761M Apr 15  2018 4R041-L7-P06-GCCAAT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  859M Apr 15  2018 4R041-L7-P06-GCCAAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 15  2018 4R041-L7-P06-GCCAAT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P07-AGTCAA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 16  2018 4R041-L7-P07-AGTCAA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 16  2018 4R041-L7-P07-AGTCAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P07-AGTCAA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P08-CTTGTA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 16  2018 4R041-L7-P08-CTTGTA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.4G Apr 16  2018 4R041-L7-P08-CTTGTA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P08-CTTGTA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 16  2018 4R041-L7-P09-GTGAAA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 1022M Apr 16  2018 4R041-L7-P09-GTGAAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  955M Apr 16  2018 4R041-L7-P09-GTGAAA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P09-GTGAAA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P10-TAGCTT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.4G Apr 16  2018 4R041-L7-P10-TAGCTT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.6G Apr 16  2018 4R041-L7-P10-TAGCTT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P10-TAGCTT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P11-ACTTGA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  916M Apr 16  2018 4R041-L7-P11-ACTTGA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 16  2018 4R041-L7-P11-ACTTGA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 16  2018 4R041-L7-P11-ACTTGA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P12-TGACCA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 16  2018 4R041-L7-P12-TGACCA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.3G Apr 16  2018 4R041-L7-P12-TGACCA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P12-TGACCA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P13-ATCACG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  760M Apr 16  2018 4R041-L7-P13-ATCACG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  786M Apr 16  2018 4R041-L7-P13-ATCACG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 16  2018 4R041-L7-P13-ATCACG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P14-GTGGCC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.4G Apr 16  2018 4R041-L7-P14-GTGGCC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.5G Apr 16  2018 4R041-L7-P14-GTGGCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P14-GTGGCC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 16  2018 4R041-L7-P15-GTTTCG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.5G Apr 16  2018 4R041-L7-P15-GTTTCG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.6G Apr 16  2018 4R041-L7-P15-GTTTCG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P15-GTTTCG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  220K Apr 16  2018 stderr.log\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:10 bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:11 bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:12 bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:13 bleached_K5_04\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:14 bleached_K5_05\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:15 bleached_K5_06\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:15 bleached_44_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:16 bleached_44_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:18 bleached_44_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:19 non-bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:20 non-bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:21 non-bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:22 non-bleached_44_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:23 non-bleached_44_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:24 non-bleached_44_03\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -ltrh /mnt/data/coral_RNAseq_2017/porites/20180311_fastqc_trimming/trimmed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 204K\n",
      "-rw-rw-r-- 1 sam sam    0 Jan 30 21:09 20190130_salmon.out\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:10 bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:11 bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:12 bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:13 bleached_K5_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:14 bleached_K5_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:15 bleached_K5_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:15 bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:16 bleached_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:18 bleached_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:19 non-bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:20 non-bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:21 non-bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:22 non-bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:23 non-bleached_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Jan 30 21:24 non-bleached_44_03\n",
      "-rw-rw-r-- 1 sam sam 138K Jan 30 21:24 20190130_salmon.err\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mv /mnt/data/coral_RNAseq_2017/porites/20180311_fastqc_trimming/trimmed/bleached* \\\n",
    "${salmon_out_dir}\n",
    "\n",
    "mv /mnt/data/coral_RNAseq_2017/porites/20180311_fastqc_trimming/trimmed/non* \\\n",
    "${salmon_out_dir}\n",
    "\n",
    "ls -ltrh ${salmon_out_dir}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 2.1M\n",
      "-rw-rw-r-- 1 sam sam     0 Jan 30 21:09 20190130_salmon.out\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:10 bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:11 bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:12 bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:13 bleached_K5_04\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:14 bleached_K5_05\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:15 bleached_K5_06\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:15 bleached_44_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:16 bleached_44_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:18 bleached_44_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:19 non-bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:20 non-bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:21 non-bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:22 non-bleached_44_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:23 non-bleached_44_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:24 non-bleached_44_03\n",
      "-rw-rw-r-- 1 sam sam  138K Jan 30 21:24 20190130_salmon.err\n",
      "-rw-rw-r-- 1 sam sam  858K Jan 31 07:21 salmon.isoform.TPM.not_cross_norm\n",
      "-rw-rw-r-- 1 sam sam 1014K Jan 31 07:21 salmon.isoform.counts.matrix\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-reading file: bleached_44_02/quant.sf\n",
      "-reading file: bleached_K5_03/quant.sf\n",
      "-reading file: bleached_K5_01/quant.sf\n",
      "-reading file: non-bleached_K5_01/quant.sf\n",
      "-reading file: non-bleached_44_03/quant.sf\n",
      "-reading file: bleached_K5_05/quant.sf\n",
      "-reading file: bleached_K5_02/quant.sf\n",
      "-reading file: non-bleached_44_02/quant.sf\n",
      "-reading file: non-bleached_44_01/quant.sf\n",
      "-reading file: bleached_44_03/quant.sf\n",
      "-reading file: bleached_K5_04/quant.sf\n",
      "-reading file: non-bleached_K5_02/quant.sf\n",
      "-reading file: non-bleached_K5_03/quant.sf\n",
      "-reading file: bleached_K5_06/quant.sf\n",
      "-reading file: bleached_44_01/quant.sf\n",
      "\n",
      "\n",
      "* Outputting combined matrix.\n",
      "\n",
      "Error, cannot open file /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.coral.fasta.gene_trans_map at /home/shared/Trinityrnaseq-v2.6.6/util/abundance_estimates_to_matrix.pl line 299.\n",
      "\n",
      "real\t0m0.617s\n",
      "user\t0m0.592s\n",
      "sys\t0m0.024s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ${salmon_out_dir}\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/util/abundance_estimates_to_matrix.pl \\\n",
    "--est_method salmon \\\n",
    "--gene_trans_map ${gene_map} \\\n",
    "--out_prefix salmon \\\n",
    "--name_sample_by_basedir \\\n",
    "bleached_44_02/quant.sf \\\n",
    "bleached_K5_03/quant.sf \\\n",
    "bleached_K5_01/quant.sf \\\n",
    "non-bleached_K5_01/quant.sf \\\n",
    "non-bleached_44_03/quant.sf \\\n",
    "bleached_K5_05/quant.sf \\\n",
    "bleached_K5_02/quant.sf \\\n",
    "non-bleached_44_02/quant.sf \\\n",
    "non-bleached_44_01/quant.sf \\\n",
    "bleached_44_03/quant.sf \\\n",
    "bleached_K5_04/quant.sf \\\n",
    "non-bleached_K5_02/quant.sf \\\n",
    "non-bleached_K5_03/quant.sf \\\n",
    "bleached_K5_06/quant.sf \\\n",
    "bleached_44_01/quant.sf\n",
    "\n",
    "ls -ltrh\n",
    "\n",
    "sed '/^Subject:/ s/ / porites_abundances JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 4.0M\n",
      "-rw-rw-r-- 1 sam sam     0 Jan 30 21:09 20190130_salmon.out\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:10 bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:11 bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:12 bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:13 bleached_K5_04\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:14 bleached_K5_05\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:15 bleached_K5_06\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:15 bleached_44_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:16 bleached_44_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:18 bleached_44_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:19 non-bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:20 non-bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:21 non-bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:22 non-bleached_44_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:23 non-bleached_44_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Jan 30 21:24 non-bleached_44_03\n",
      "-rw-rw-r-- 1 sam sam  138K Jan 30 21:24 20190130_salmon.err\n",
      "-rw-rw-r-- 1 sam sam  858K Jan 31 07:22 salmon.isoform.TPM.not_cross_norm\n",
      "-rw-rw-r-- 1 sam sam 1014K Jan 31 07:22 salmon.isoform.counts.matrix\n",
      "-rw-rw-r-- 1 sam sam  492K Jan 31 07:22 salmon.gene.TPM.not_cross_norm\n",
      "-rw-rw-r-- 1 sam sam  348K Jan 31 07:22 salmon.gene.counts.matrix\n",
      "-rw-rw-r-- 1 sam sam   528 Jan 31 07:22 salmon.isoform.TPM.not_cross_norm.runTMM.R\n",
      "-rw-rw-r-- 1 sam sam   914 Jan 31 07:22 salmon.isoform.TPM.not_cross_norm.TMM_info.txt\n",
      "-rw-rw-r-- 1 sam sam  733K Jan 31 07:22 salmon.isoform.TMM.EXPR.matrix\n",
      "-rw-rw-r-- 1 sam sam   522 Jan 31 07:22 salmon.gene.TPM.not_cross_norm.runTMM.R\n",
      "-rw-rw-r-- 1 sam sam   917 Jan 31 07:22 salmon.gene.TPM.not_cross_norm.TMM_info.txt\n",
      "-rw-rw-r-- 1 sam sam  377K Jan 31 07:22 salmon.gene.TMM.EXPR.matrix\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-reading file: bleached_44_02/quant.sf\n",
      "-reading file: bleached_K5_03/quant.sf\n",
      "-reading file: bleached_K5_01/quant.sf\n",
      "-reading file: non-bleached_K5_01/quant.sf\n",
      "-reading file: non-bleached_44_03/quant.sf\n",
      "-reading file: bleached_K5_05/quant.sf\n",
      "-reading file: bleached_K5_02/quant.sf\n",
      "-reading file: non-bleached_44_02/quant.sf\n",
      "-reading file: non-bleached_44_01/quant.sf\n",
      "-reading file: bleached_44_03/quant.sf\n",
      "-reading file: bleached_K5_04/quant.sf\n",
      "-reading file: non-bleached_K5_02/quant.sf\n",
      "-reading file: non-bleached_K5_03/quant.sf\n",
      "-reading file: bleached_K5_06/quant.sf\n",
      "-reading file: bleached_44_01/quant.sf\n",
      "\n",
      "\n",
      "* Outputting combined matrix.\n",
      "\n",
      "/home/shared/Trinityrnaseq-v2.6.6/util/support_scripts/run_TMM_scale_matrix.pl --matrix salmon.isoform.TPM.not_cross_norm > salmon.isoform.TMM.EXPR.matrixCMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.TPM.not_cross_norm.runTMM.R 1>&2 \n",
      "> library(edgeR)\n",
      "Loading required package: limma\n",
      "> \n",
      "> rnaseqMatrix = read.table(\"salmon.isoform.TPM.not_cross_norm\", header=T, row.names=1, com='', check.names=F)\n",
      "> rnaseqMatrix = as.matrix(rnaseqMatrix)\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=factor(colnames(rnaseqMatrix)))\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study$samples$eff.lib.size = exp_study$samples$lib.size * exp_study$samples$norm.factors\n",
      "> write.table(exp_study$samples, file=\"salmon.isoform.TPM.not_cross_norm.TMM_info.txt\", quote=F, sep=\"\\t\", row.names=F)\n",
      "> \n",
      "/home/shared/Trinityrnaseq-v2.6.6/util/support_scripts/run_TMM_scale_matrix.pl --matrix salmon.gene.TPM.not_cross_norm > salmon.gene.TMM.EXPR.matrixCMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.gene.TPM.not_cross_norm.runTMM.R 1>&2 \n",
      "> library(edgeR)\n",
      "Loading required package: limma\n",
      "> \n",
      "> rnaseqMatrix = read.table(\"salmon.gene.TPM.not_cross_norm\", header=T, row.names=1, com='', check.names=F)\n",
      "> rnaseqMatrix = as.matrix(rnaseqMatrix)\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=factor(colnames(rnaseqMatrix)))\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study$samples$eff.lib.size = exp_study$samples$lib.size * exp_study$samples$norm.factors\n",
      "> write.table(exp_study$samples, file=\"salmon.gene.TPM.not_cross_norm.TMM_info.txt\", quote=F, sep=\"\\t\", row.names=F)\n",
      "> \n",
      "\n",
      "real\t0m3.105s\n",
      "user\t0m2.996s\n",
      "sys\t0m0.084s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ${salmon_out_dir}\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/util/abundance_estimates_to_matrix.pl \\\n",
    "--est_method salmon \\\n",
    "--gene_trans_map ${gene_map} \\\n",
    "--out_prefix salmon \\\n",
    "--name_sample_by_basedir \\\n",
    "bleached_44_02/quant.sf \\\n",
    "bleached_K5_03/quant.sf \\\n",
    "bleached_K5_01/quant.sf \\\n",
    "non-bleached_K5_01/quant.sf \\\n",
    "non-bleached_44_03/quant.sf \\\n",
    "bleached_K5_05/quant.sf \\\n",
    "bleached_K5_02/quant.sf \\\n",
    "non-bleached_44_02/quant.sf \\\n",
    "non-bleached_44_01/quant.sf \\\n",
    "bleached_44_03/quant.sf \\\n",
    "bleached_K5_04/quant.sf \\\n",
    "non-bleached_K5_02/quant.sf \\\n",
    "non-bleached_K5_03/quant.sf \\\n",
    "bleached_K5_06/quant.sf \\\n",
    "bleached_44_01/quant.sf\n",
    "\n",
    "ls -ltrh\n",
    "\n",
    "sed '/^Subject:/ s/ / porites_abundances JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bleached_44_02             bleached_K5_03  bleached_K5_01  non-bleached_K5_01  non-bleached_44_03  bleached_K5_05  bleached_K5_02  non-bleached_44_02  non-bleached_44_01  bleached_44_03  bleached_K5_04  non-bleached_K5_02  non-bleached_K5_03  bleached_K5_06  bleached_44_01\n",
      "TRINITY_DN163876_c0_g1_i1  0.000000        7.940706        0.000000            0.000000            0.000000        0.000000        0.000000            12.189162           0.000000        6.194858        0.000000            7.257273            12.187844       0.000000        0.000000\n",
      "TRINITY_DN185426_c2_g3_i2  178.883547      66.488310       97.342583           56.752588           100.990795      244.000000      81.000000           65.090225           52.821492       82.988059       204.000000          121.496359          211.233811      35.449484       72.997456\n",
      "TRINITY_DN146585_c0_g1_i1  11.000000       7.000000        24.000000           13.000000           33.000000       2.000000        17.000000           16.000000           23.000000       47.000000       19.000000           64.000000           15.000000       12.000000       42.000000\n",
      "TRINITY_DN182830_c5_g1_i4  0.000000        2373.770982     31.968865           17.064769           23.441255       87.231155       0.000000            0.000000            0.000000        38.274316       0.000000            0.000000            0.000000        0.000000        0.000000\n",
      "TRINITY_DN196450_c1_g1_i2  99.555832       259.953213      79.961908           0.000000            280.008700      20.155948       341.471229          299.274216          220.000000      108.211215      42.626532           0.000000            89.139854       117.612350      170.002470\n",
      "TRINITY_DN137809_c0_g2_i2  111.886433      174.053920      151.083461          188.625312          275.000000      4.072560        29.380098           80.597720           212.161748      231.774227      237.597315          546.836092          97.596599       91.679369       315.394252\n",
      "TRINITY_DN191294_c2_g1_i5  6.744369        0.000000        5.689976            10.813362           3.442361        0.000000        0.000000            6.824969            6.351809        0.000000        0.000000            0.000000            0.000000        0.000000        0.000000\n",
      "TRINITY_DN195629_c3_g1_i1  110.000000      241.000000      30.000000           71.000000           16.000000       50.000000       14.000000           7.000000            86.000000       25.000000       40.000000           1094.000000         77.000000       13.000000       61.000000\n",
      "TRINITY_DN190496_c0_g2_i1  0.000000        0.000000        0.000000            213.939364          0.000000        0.000000        80.956047           146.128402          0.000000        0.000000        0.000000            0.000000            0.000000        0.000000        0.000000\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "head ${salmon_out_dir}/salmon.isoform.counts.matrix \\\n",
    "| column -t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "$VAR1 = {\n",
      "          'non-bleached' => [\n",
      "                              'non-bleached_K5_01',\n",
      "                              'non-bleached_K5_02',\n",
      "                              'non-bleached_K5_03',\n",
      "                              'non-bleached_44_01',\n",
      "                              'non-bleached_44_02',\n",
      "                              'non-bleached_44_03'\n",
      "                            ],\n",
      "          'bleached' => [\n",
      "                          'bleached_K5_01',\n",
      "                          'bleached_K5_02',\n",
      "                          'bleached_K5_03',\n",
      "                          'bleached_K5_04',\n",
      "                          'bleached_K5_05',\n",
      "                          'bleached_K5_06',\n",
      "                          'bleached_44_01',\n",
      "                          'bleached_44_02',\n",
      "                          'bleached_44_03'\n",
      "                        ]\n",
      "        };\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.bleached_vs_non-bleached.bleached.vs.non-bleached.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/porites/20190130_salmon/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(3,7,2,11,6,14,15,1,10,4,12,13,9,8,5)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"bleached\", 9), rep(\"non-bleached\", 6)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"bleached\", \"non-bleached\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"bleached\", sampleB=\"non-bleached\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Got 15 samples, and got: 16 data fields.\n",
      "Header: bleached_44_02\tbleached_K5_03\tbleached_K5_01\tnon-bleached_K5_01\tnon-bleached_44_03\tbleached_K5_05\tbleached_K5_02\tnon-bleached_44_02\tnon-bleached_44_01\tbleached_44_03\tbleached_K5_04\tnon-bleached_K5_02\tnon-bleached_K5_03\tbleached_K5_06\tbleached_44_01\n",
      "Next: TRINITY_DN163876_c0_g1_i1\t0.000000\t7.940706\t0.000000\t0.000000\t0.000000\t0.000000\t0.000000\t12.189162\t0.000000\t6.194858\t0.000000\t7.257273\t12.187844\t0.000000\t0.000000\n",
      "\n",
      "$VAR1 = {\n",
      "          'bleached_K5_05' => 6,\n",
      "          'non-bleached_44_01' => 9,\n",
      "          'bleached_K5_03' => 2,\n",
      "          'non-bleached_K5_01' => 4,\n",
      "          'non-bleached_44_02' => 8,\n",
      "          'non-bleached_44_03' => 5,\n",
      "          'bleached_44_03' => 10,\n",
      "          'non-bleached_K5_03' => 13,\n",
      "          'bleached_K5_06' => 14,\n",
      "          'bleached_44_02' => 1,\n",
      "          'bleached_K5_04' => 11,\n",
      "          'bleached_44_01' => 15,\n",
      "          'bleached_K5_02' => 7,\n",
      "          'non-bleached_K5_02' => 12,\n",
      "          'bleached_K5_01' => 3\n",
      "        };\n",
      "Contrasts to perform are: $VAR1 = [\n",
      "          [\n",
      "            'bleached',\n",
      "            'non-bleached'\n",
      "          ]\n",
      "        ];\n",
      "Loading required package: limma\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts\n",
    "\n",
    "/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/run_DE_analysis.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.counts.matrix \\\n",
    "--method edgeR \\\n",
    "--samples_file ${samples} \n",
    "\n",
    "sed '/^Subject:/ s/ / porites_edgeR JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> library(cluster)\n",
      "> library(Biobase)\n",
      "> library(qvalue)\n",
      "> library(fastcluster)\n",
      "> options(stringsAsFactors = FALSE)\n",
      "> NO_REUSE = F\n",
      "> \n",
      "> # try to reuse earlier-loaded data if possible\n",
      "> if (file.exists(\"diffExpr.P0.001_C2.matrix.RData\") && ! NO_REUSE) {\n",
      "+     print('RESTORING DATA FROM EARLIER ANALYSIS')\n",
      "+     load(\"diffExpr.P0.001_C2.matrix.RData\")\n",
      "+ } else {\n",
      "+     print('Reading matrix file.')\n",
      "+     primary_data = read.table(\"diffExpr.P0.001_C2.matrix\", header=T, com='', row.names=1, check.names=F)\n",
      "+     primary_data = as.matrix(primary_data)\n",
      "+ }\n",
      "[1] \"Reading matrix file.\"\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/heatmap.3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/misc_rnaseq_funcs.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/pairs3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/vioplot2.R\")\n",
      "> data = primary_data\n",
      "> myheatcol = colorpanel(75, 'purple','black','yellow')\n",
      "> samples_data = read.table(\"/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_trinity_sample_list.txt\", header=F, check.names=F, fill=T)\n",
      "> samples_data = samples_data[samples_data[,2] != '',]\n",
      "> colnames(samples_data) = c('sample_name', 'replicate_name')\n",
      "> sample_types = as.character(unique(samples_data[,1]))\n",
      "> rep_names = as.character(samples_data[,2])\n",
      "> data = data[, colnames(data) %in% rep_names, drop=F ]\n",
      "> nsamples = length(sample_types)\n",
      "> sample_colors = rainbow(nsamples)\n",
      "> names(sample_colors) = sample_types\n",
      "> sample_type_list = list()\n",
      "> for (i in 1:nsamples) {\n",
      "+     samples_want = samples_data[samples_data[,1]==sample_types[i], 2]\n",
      "+     sample_type_list[[sample_types[i]]] = as.vector(samples_want)\n",
      "+ }\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> initial_matrix = data # store before doing various data transformations\n",
      "> data = log2(data+1)\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> sampleAnnotations = matrix(ncol=ncol(data),nrow=nsamples)\n",
      "> for (i in 1:nsamples) {\n",
      "+   sampleAnnotations[i,] = colnames(data) %in% sample_type_list[[sample_types[i]]]\n",
      "+ }\n",
      "> sampleAnnotations = apply(sampleAnnotations, 1:2, function(x) as.logical(x))\n",
      "> sampleAnnotations = sample_matrix_to_color_assignments(sampleAnnotations, col=sample_colors)\n",
      "> rownames(sampleAnnotations) = as.vector(sample_types)\n",
      "> colnames(sampleAnnotations) = colnames(data)\n",
      "> data = as.matrix(data) # convert to matrix\n",
      "> \n",
      "> # Centering rows\n",
      "> data = t(scale(t(data), scale=F))\n",
      "> \n",
      "> write.table(data, file=\"diffExpr.P0.001_C2.matrix.log2.centered.dat\", quote=F, sep='\t');\n",
      "> if (nrow(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two rows are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> if (ncol(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two columns are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> sample_cor = cor(data, method='pearson', use='pairwise.complete.obs')\n",
      "> write.table(sample_cor, file=\"diffExpr.P0.001_C2.matrix.log2.centered.sample_cor.dat\", quote=F, sep='\t')\n",
      "> sample_dist = dist(t(data), method='euclidean')\n",
      "> hc_samples = hclust(sample_dist, method='complete')\n",
      "> pdf(\"diffExpr.P0.001_C2.matrix.log2.centered.sample_cor_matrix.pdf\")\n",
      "> sample_cor_for_plot = sample_cor\n",
      "> heatmap.3(sample_cor_for_plot, dendrogram='both', Rowv=as.dendrogram(hc_samples), Colv=as.dendrogram(hc_samples), col = myheatcol, scale='none', symm=TRUE, key=TRUE,density.info='none', trace='none', symkey=FALSE, symbreaks=F, margins=c(10,10), cexCol=1, cexRow=1, cex.main=0.75, main=paste(\"sample correlation matrix\n",
      "+ \", \"diffExpr.P0.001_C2.matrix.log2.centered\") , ColSideColors=sampleAnnotations, RowSideColors=t(sampleAnnotations))\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> gene_cor = NULL\n",
      "> gene_dist = dist(data, method='euclidean')\n",
      "> if (nrow(data) <= 1) { message('Too few genes to generate heatmap'); quit(status=0); }\n",
      "> hc_genes = hclust(gene_dist, method='complete')\n",
      "> heatmap_data = data\n",
      "> pdf(\"diffExpr.P0.001_C2.matrix.log2.centered.genes_vs_samples_heatmap.pdf\")\n",
      "> heatmap.3(heatmap_data, dendrogram='both', Rowv=as.dendrogram(hc_genes), Colv=as.dendrogram(hc_samples), col=myheatcol, scale=\"none\", density.info=\"none\", trace=\"none\", key=TRUE, keysize=1.2, cexCol=1, margins=c(10,10), cex.main=0.75, main=paste(\"samples vs. features\n",
      "+ \", \"diffExpr.P0.001_C2.matrix.log2.centered\" ) , ColSideColors=sampleAnnotations)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> save(list=ls(all=TRUE), file=\"diffExpr.P0.001_C2.matrix.RData\")\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "** Found 3 features as differentially expressed.\n",
      "\n",
      "CMD: /home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/PtR -m diffExpr.P0.001_C2.matrix --log2 --heatmap --min_colSums 0 --min_rowSums 0 --gene_dist euclidean --sample_dist euclidean --sample_cor_matrix --center_rows --save  -s /home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_trinity_sample_list.txt\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < diffExpr.P0.001_C2.matrix.R\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘fastcluster’\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    hclust\n",
      "\n",
      "for plotting:: min.raw: -0.993529364990907 max.raw: 1\n",
      "for plotting:: min.raw: -8.55530835383169 max.raw: 8.55530835383169\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/\n",
    "\n",
    "/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/analyze_diff_expr.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.TMM.EXPR.matrix \\\n",
    "--samples ${samples}\n",
    "\n",
    "sed '/^Subject:/ s/ / porites_edgeR JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 /media/sam/4TB_toshiba/porites/coral_transcripts/edgeR.11860.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.001_C2.bleached-UP.subset\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/\n",
    "\n",
    "wc -l /media/sam/4TB_toshiba/porites/coral_transcripts/edgeR.11860.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.001_C2.bleached-UP.subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 /media/sam/4TB_toshiba/porites/coral_transcripts/edgeR.11860.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.001_C2.non-bleached-UP.subset\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/\n",
    "\n",
    "wc -l /media/sam/4TB_toshiba/porites/coral_transcripts/edgeR.11860.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.001_C2.non-bleached-UP.subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> library(cluster)\n",
      "> library(Biobase)\n",
      "> library(qvalue)\n",
      "> library(fastcluster)\n",
      "> options(stringsAsFactors = FALSE)\n",
      "> NO_REUSE = F\n",
      "> \n",
      "> # try to reuse earlier-loaded data if possible\n",
      "> if (file.exists(\"diffExpr.P0.05_C1.matrix.RData\") && ! NO_REUSE) {\n",
      "+     print('RESTORING DATA FROM EARLIER ANALYSIS')\n",
      "+     load(\"diffExpr.P0.05_C1.matrix.RData\")\n",
      "+ } else {\n",
      "+     print('Reading matrix file.')\n",
      "+     primary_data = read.table(\"diffExpr.P0.05_C1.matrix\", header=T, com='', row.names=1, check.names=F)\n",
      "+     primary_data = as.matrix(primary_data)\n",
      "+ }\n",
      "[1] \"Reading matrix file.\"\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/heatmap.3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/misc_rnaseq_funcs.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/pairs3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/vioplot2.R\")\n",
      "> data = primary_data\n",
      "> myheatcol = colorpanel(75, 'purple','black','yellow')\n",
      "> samples_data = read.table(\"/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_trinity_sample_list.txt\", header=F, check.names=F, fill=T)\n",
      "> samples_data = samples_data[samples_data[,2] != '',]\n",
      "> colnames(samples_data) = c('sample_name', 'replicate_name')\n",
      "> sample_types = as.character(unique(samples_data[,1]))\n",
      "> rep_names = as.character(samples_data[,2])\n",
      "> data = data[, colnames(data) %in% rep_names, drop=F ]\n",
      "> nsamples = length(sample_types)\n",
      "> sample_colors = rainbow(nsamples)\n",
      "> names(sample_colors) = sample_types\n",
      "> sample_type_list = list()\n",
      "> for (i in 1:nsamples) {\n",
      "+     samples_want = samples_data[samples_data[,1]==sample_types[i], 2]\n",
      "+     sample_type_list[[sample_types[i]]] = as.vector(samples_want)\n",
      "+ }\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> initial_matrix = data # store before doing various data transformations\n",
      "> data = log2(data+1)\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> sampleAnnotations = matrix(ncol=ncol(data),nrow=nsamples)\n",
      "> for (i in 1:nsamples) {\n",
      "+   sampleAnnotations[i,] = colnames(data) %in% sample_type_list[[sample_types[i]]]\n",
      "+ }\n",
      "> sampleAnnotations = apply(sampleAnnotations, 1:2, function(x) as.logical(x))\n",
      "> sampleAnnotations = sample_matrix_to_color_assignments(sampleAnnotations, col=sample_colors)\n",
      "> rownames(sampleAnnotations) = as.vector(sample_types)\n",
      "> colnames(sampleAnnotations) = colnames(data)\n",
      "> data = as.matrix(data) # convert to matrix\n",
      "> \n",
      "> # Centering rows\n",
      "> data = t(scale(t(data), scale=F))\n",
      "> \n",
      "> write.table(data, file=\"diffExpr.P0.05_C1.matrix.log2.centered.dat\", quote=F, sep='\t');\n",
      "> if (nrow(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two rows are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> if (ncol(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two columns are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> sample_cor = cor(data, method='pearson', use='pairwise.complete.obs')\n",
      "> write.table(sample_cor, file=\"diffExpr.P0.05_C1.matrix.log2.centered.sample_cor.dat\", quote=F, sep='\t')\n",
      "> sample_dist = dist(t(data), method='euclidean')\n",
      "> hc_samples = hclust(sample_dist, method='complete')\n",
      "> pdf(\"diffExpr.P0.05_C1.matrix.log2.centered.sample_cor_matrix.pdf\")\n",
      "> sample_cor_for_plot = sample_cor\n",
      "> heatmap.3(sample_cor_for_plot, dendrogram='both', Rowv=as.dendrogram(hc_samples), Colv=as.dendrogram(hc_samples), col = myheatcol, scale='none', symm=TRUE, key=TRUE,density.info='none', trace='none', symkey=FALSE, symbreaks=F, margins=c(10,10), cexCol=1, cexRow=1, cex.main=0.75, main=paste(\"sample correlation matrix\n",
      "+ \", \"diffExpr.P0.05_C1.matrix.log2.centered\") , ColSideColors=sampleAnnotations, RowSideColors=t(sampleAnnotations))\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> gene_cor = NULL\n",
      "> gene_dist = dist(data, method='euclidean')\n",
      "> if (nrow(data) <= 1) { message('Too few genes to generate heatmap'); quit(status=0); }\n",
      "> hc_genes = hclust(gene_dist, method='complete')\n",
      "> heatmap_data = data\n",
      "> pdf(\"diffExpr.P0.05_C1.matrix.log2.centered.genes_vs_samples_heatmap.pdf\")\n",
      "> heatmap.3(heatmap_data, dendrogram='both', Rowv=as.dendrogram(hc_genes), Colv=as.dendrogram(hc_samples), col=myheatcol, scale=\"none\", density.info=\"none\", trace=\"none\", key=TRUE, keysize=1.2, cexCol=1, margins=c(10,10), cex.main=0.75, main=paste(\"samples vs. features\n",
      "+ \", \"diffExpr.P0.05_C1.matrix.log2.centered\" ) , ColSideColors=sampleAnnotations)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> save(list=ls(all=TRUE), file=\"diffExpr.P0.05_C1.matrix.RData\")\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "** Found 17 features as differentially expressed.\n",
      "\n",
      "CMD: /home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/PtR -m diffExpr.P0.05_C1.matrix --log2 --heatmap --min_colSums 0 --min_rowSums 0 --gene_dist euclidean --sample_dist euclidean --sample_cor_matrix --center_rows --save  -s /home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_trinity_sample_list.txt\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < diffExpr.P0.05_C1.matrix.R\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘fastcluster’\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    hclust\n",
      "\n",
      "for plotting:: min.raw: -0.6324978111069 max.raw: 1\n",
      "for plotting:: min.raw: -9.75042604057765 max.raw: 9.75042604057765\n",
      "\n",
      "real\t0m1.708s\n",
      "user\t0m1.640s\n",
      "sys\t0m0.056s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/analyze_diff_expr.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.TMM.EXPR.matrix \\\n",
    "--samples ${samples} \\\n",
    "-C 1 \\\n",
    "-P 0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4-fold bleached up, 0.001 p value\n",
      "1 /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.001_C2.bleached-UP.subset\n",
      "2-fold bleached up, 0.05 p value\n",
      "5 /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.05_C1.bleached-UP.subset\n",
      "4-fold non-bleached up, 0.001 p value\n",
      "4 /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.001_C2.non-bleached-UP.subset\n",
      "2-fold non-bleached up, 0.05 p value\n",
      "14 /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.05_C1.non-bleached-UP.subset\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/\n",
    "\n",
    "echo \"4-fold bleached up, 0.001 p value\"\n",
    "wc -l /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.001_C2.bleached-UP.subset\n",
    "\n",
    "echo \"2-fold bleached up, 0.05 p value\"\n",
    "wc -l /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.05_C1.bleached-UP.subset\n",
    "\n",
    "echo \"4-fold non-bleached up, 0.001 p value\"\n",
    "wc -l /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.001_C2.non-bleached-UP.subset\n",
    "\n",
    "echo \"2-fold non-bleached up, 0.05 p value\"\n",
    "wc -l /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.3209.dir/salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.P0.05_C1.non-bleached-UP.subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
