{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TODAY'S DATE\n",
      "Tue Mar  5 23:06:25 PST 2019\n",
      "------------\n",
      "\n",
      "Distributor ID:\tUbuntu\n",
      "Description:\tUbuntu 16.04.5 LTS\n",
      "Release:\t16.04\n",
      "Codename:\txenial\n",
      "\n",
      "------------\n",
      "HOSTNAME: \n",
      "jpg-d20\n",
      "\n",
      "------------\n",
      "Computer Specs:\n",
      "\n",
      "Architecture:          x86_64\n",
      "CPU op-mode(s):        32-bit, 64-bit\n",
      "Byte Order:            Little Endian\n",
      "CPU(s):                24\n",
      "On-line CPU(s) list:   0-23\n",
      "Thread(s) per core:    2\n",
      "Core(s) per socket:    6\n",
      "Socket(s):             2\n",
      "NUMA node(s):          2\n",
      "Vendor ID:             GenuineIntel\n",
      "CPU family:            6\n",
      "Model:                 44\n",
      "Model name:            Intel(R) Xeon(R) CPU           X5680  @ 3.33GHz\n",
      "Stepping:              2\n",
      "CPU MHz:               1596.000\n",
      "CPU max MHz:           3326.0000\n",
      "CPU min MHz:           1596.0000\n",
      "BogoMIPS:              6666.90\n",
      "Virtualization:        VT-x\n",
      "L1d cache:             32K\n",
      "L1i cache:             32K\n",
      "L2 cache:              256K\n",
      "L3 cache:              12288K\n",
      "NUMA node0 CPU(s):     0,2,4,6,8,10,12,14,16,18,20,22\n",
      "NUMA node1 CPU(s):     1,3,5,7,9,11,13,15,17,19,21,23\n",
      "Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 popcnt aes lahf_lm epb kaiser tpr_shadow vnmi flexpriority ept vpid dtherm ida arat\n",
      "\n",
      "------------\n",
      "\n",
      "Memory Specs\n",
      "\n",
      "              total        used        free      shared  buff/cache   available\n",
      "Mem:            94G        2.7G        2.5G        1.0G         89G         90G\n",
      "Swap:          128G          0B        128G\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No LSB modules are available.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"TODAY'S DATE\"\n",
    "date\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "lsb_release -a\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"HOSTNAME: \"\n",
    "hostname\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"Computer Specs:\"\n",
    "echo \"\"\n",
    "lscpu\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "echo \"Memory Specs\"\n",
    "echo \"\"\n",
    "free -mh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: salmon_out_dir=/media/sam/4TB_toshiba/porites/20190305_salmon_abundance_porites\n",
      "env: coral_transcriptome=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta\n",
      "env: coral_fasta_index=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.fai\n",
      "env: samples=/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_44_vs_k5_trinity_sample_list.txt\n",
      "env: gene_map=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.gene_trans_map\n",
      "env: salmon_stdout=20190305_salmon.out\n",
      "env: salmon_stderr=20190305_salmon.err\n"
     ]
    }
   ],
   "source": [
    "%env salmon_out_dir=/media/sam/4TB_toshiba/porites/20190305_salmon_abundance_porites\n",
    "%env coral_transcriptome=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta\n",
    "%env coral_fasta_index=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.fai\n",
    "%env samples=/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_44_vs_k5_trinity_sample_list.txt\n",
    "%env gene_map=/media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.gene_trans_map\n",
    "\n",
    "%env salmon_stdout=20190305_salmon.out\n",
    "%env salmon_stderr=20190305_salmon.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bleached  bleached_K5_01  4R041-L7-P01-AGTTCC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P01-AGTTCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_K5_02  4R041-L7-P06-GCCAAT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P06-GCCAAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_K5_03  4R041-L7-P07-AGTCAA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P07-AGTCAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_K5_04  4R041-L7-P08-CTTGTA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P08-CTTGTA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_K5_05  4R041-L7-P09-GTGAAA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P09-GTGAAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_K5_06  4R041-L7-P13-ATCACG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P13-ATCACG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_44_01  4R041-L7-P03-ACTGAT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P03-ACTGAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_44_02  4R041-L7-P11-ACTTGA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P11-ACTTGA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_44_03  4R041-L7-P14-GTGGCC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P14-GTGGCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_K5_01  4R041-L7-P02-ACAGTG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P02-ACAGTG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_K5_02  4R041-L7-P10-TAGCTT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P10-TAGCTT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_K5_03  4R041-L7-P15-GTTTCG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P15-GTTTCG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_44_01  4R041-L7-P04-GTCCGC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P04-GTCCGC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_44_02  4R041-L7-P05-GAGTGG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P05-GAGTGG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_44_03  4R041-L7-P12-TGACCA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P12-TGACCA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "44  b_44_01  4R041-L7-P03-ACTGAT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P03-ACTGAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "44  b_44_02  4R041-L7-P11-ACTTGA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P11-ACTTGA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "44  b_44_03  4R041-L7-P14-GTGGCC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P14-GTGGCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "44  nb_44_04  4R041-L7-P04-GTCCGC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P04-GTCCGC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "44  nb_44_05  4R041-L7-P05-GAGTGG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P05-GAGTGG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "44  nb_44_06  4R041-L7-P12-TGACCA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P12-TGACCA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k5  b_K5_01  4R041-L7-P01-AGTTCC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P01-AGTTCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k5  b_K5_02  4R041-L7-P06-GCCAAT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P06-GCCAAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k5  b_K5_03  4R041-L7-P07-AGTCAA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P07-AGTCAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k5  b_K5_04  4R041-L7-P08-CTTGTA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P08-CTTGTA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k5  b_K5_05  4R041-L7-P09-GTGAAA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P09-GTGAAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k5  b_K5_06  4R041-L7-P13-ATCACG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P13-ATCACG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k5  nb_K5_07  4R041-L7-P02-ACAGTG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P02-ACAGTG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k5  nb_K5_08  4R041-L7-P10-TAGCTT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P10-TAGCTT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k5  nbK5_09  4R041-L7-P15-GTTTCG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L7-P15-GTTTCG-READ2-Sequences.txt.gz_val_2.fq.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat ${samples}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "real\t69m34.629s\n",
      "user\t655m17.032s\n",
      "sys\t7m50.392s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /mnt/data/coral_RNAseq_2017/porites/20180311_fastqc_trimming/trimmed\n",
    "\n",
    "time  /home/shared/Trinityrnaseq-v2.6.6/util/align_and_estimate_abundance.pl \\\n",
    "--output_dir ${salmon_out_dir} \\\n",
    "--transcripts ${coral_transcriptome} \\\n",
    "--seqType fq \\\n",
    "--samples_file ${samples} \\\n",
    "--SS_lib_type RF \\\n",
    "--est_method salmon \\\n",
    "--aln_method bowtie2 \\\n",
    "--trinity_mode \\\n",
    "--prep_reference \\\n",
    "--thread_count 23 \\\n",
    "1> ${salmon_out_dir}/${salmon_stdout} \\\n",
    "2> ${salmon_out_dir}/${salmon_stderr}\n",
    "sed '/^Subject:/ s/ / porites_rsem JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 34G\n",
      "drwxrwxr-x 2 sam sam  4.0K Apr 15  2018 fastqc\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P01-AGTTCC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P01-AGTTCC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 15  2018 4R041-L7-P01-AGTTCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P01-AGTTCC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 15  2018 4R041-L7-P02-ACAGTG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  998M Apr 15  2018 4R041-L7-P02-ACAGTG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 15  2018 4R041-L7-P02-ACAGTG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P02-ACAGTG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P03-ACTGAT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  956M Apr 15  2018 4R041-L7-P03-ACTGAT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P03-ACTGAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P03-ACTGAT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 15  2018 4R041-L7-P04-GTCCGC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  941M Apr 15  2018 4R041-L7-P04-GTCCGC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P04-GTCCGC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 15  2018 4R041-L7-P04-GTCCGC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 15  2018 4R041-L7-P05-GAGTGG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 15  2018 4R041-L7-P05-GAGTGG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 15  2018 4R041-L7-P05-GAGTGG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.9K Apr 15  2018 4R041-L7-P05-GAGTGG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P06-GCCAAT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  761M Apr 15  2018 4R041-L7-P06-GCCAAT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  859M Apr 15  2018 4R041-L7-P06-GCCAAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 15  2018 4R041-L7-P06-GCCAAT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 15  2018 4R041-L7-P07-AGTCAA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 16  2018 4R041-L7-P07-AGTCAA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 16  2018 4R041-L7-P07-AGTCAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P07-AGTCAA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P08-CTTGTA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.2G Apr 16  2018 4R041-L7-P08-CTTGTA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.4G Apr 16  2018 4R041-L7-P08-CTTGTA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P08-CTTGTA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 16  2018 4R041-L7-P09-GTGAAA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 1022M Apr 16  2018 4R041-L7-P09-GTGAAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  955M Apr 16  2018 4R041-L7-P09-GTGAAA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P09-GTGAAA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P10-TAGCTT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.4G Apr 16  2018 4R041-L7-P10-TAGCTT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.6G Apr 16  2018 4R041-L7-P10-TAGCTT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P10-TAGCTT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P11-ACTTGA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  916M Apr 16  2018 4R041-L7-P11-ACTTGA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 16  2018 4R041-L7-P11-ACTTGA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 16  2018 4R041-L7-P11-ACTTGA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P12-TGACCA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.1G Apr 16  2018 4R041-L7-P12-TGACCA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.3G Apr 16  2018 4R041-L7-P12-TGACCA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P12-TGACCA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P13-ATCACG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  760M Apr 16  2018 4R041-L7-P13-ATCACG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  786M Apr 16  2018 4R041-L7-P13-ATCACG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.7K Apr 16  2018 4R041-L7-P13-ATCACG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.5K Apr 16  2018 4R041-L7-P14-GTGGCC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.4G Apr 16  2018 4R041-L7-P14-GTGGCC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.5G Apr 16  2018 4R041-L7-P14-GTGGCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P14-GTGGCC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  4.6K Apr 16  2018 4R041-L7-P15-GTTTCG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  1.5G Apr 16  2018 4R041-L7-P15-GTTTCG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  1.6G Apr 16  2018 4R041-L7-P15-GTTTCG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam  4.8K Apr 16  2018 4R041-L7-P15-GTTTCG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam  220K Apr 16  2018 stderr.log\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:10 bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:12 bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:14 bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:17 bleached_K5_04\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:19 bleached_K5_05\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:20 bleached_K5_06\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:22 bleached_44_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:24 bleached_44_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:28 bleached_44_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:30 non-bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:34 non-bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:37 non-bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:39 non-bleached_44_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:42 non-bleached_44_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:45 non-bleached_44_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:47 b_44_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:49 b_44_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:52 b_44_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:54 nb_44_04\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:57 nb_44_05\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  5 23:59 nb_44_06\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  6 00:02 b_K5_01\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  6 00:04 b_K5_02\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  6 00:06 b_K5_03\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  6 00:09 b_K5_04\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  6 00:11 b_K5_05\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  6 00:13 b_K5_06\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  6 00:15 nb_K5_07\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  6 00:16 nb_K5_08\n",
      "drwxrwxr-x 5 sam sam  4.0K Mar  6 00:17 nbK5_09\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -ltrh /mnt/data/coral_RNAseq_2017/porites/20180311_fastqc_trimming/trimmed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 396K\n",
      "-rw-rw-r-- 1 sam sam    0 Mar  5 23:08 20190305_salmon.out\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:10 bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:12 bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:14 bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:17 bleached_K5_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:19 bleached_K5_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:20 bleached_K5_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:22 bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:24 bleached_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:28 bleached_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:30 non-bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:34 non-bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:37 non-bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:39 non-bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:42 non-bleached_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:45 non-bleached_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:47 b_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:49 b_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:52 b_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:54 nb_44_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:57 nb_44_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:59 nb_44_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:02 b_K5_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:04 b_K5_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:06 b_K5_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:09 b_K5_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:11 b_K5_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:13 b_K5_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:15 nb_K5_07\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:16 nb_K5_08\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:17 nbK5_09\n",
      "-rw-rw-r-- 1 sam sam 272K Mar  6 00:17 20190305_salmon.err\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mv /mnt/data/coral_RNAseq_2017/porites/20180311_fastqc_trimming/trimmed/[bn]* \\\n",
    "${salmon_out_dir}\n",
    "\n",
    "ls -ltrh ${salmon_out_dir}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 7.3M\n",
      "-rw-rw-r-- 1 sam sam    0 Mar  5 23:08 20190305_salmon.out\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:10 bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:12 bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:14 bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:17 bleached_K5_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:19 bleached_K5_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:20 bleached_K5_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:22 bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:24 bleached_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:28 bleached_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:30 non-bleached_K5_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:34 non-bleached_K5_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:37 non-bleached_K5_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:39 non-bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:42 non-bleached_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:45 non-bleached_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:47 b_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:49 b_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:52 b_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:54 nb_44_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:57 nb_44_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:59 nb_44_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:02 b_K5_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:04 b_K5_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:06 b_K5_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:09 b_K5_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:11 b_K5_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:13 b_K5_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:15 nb_K5_07\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:16 nb_K5_08\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:17 nbK5_09\n",
      "-rw-rw-r-- 1 sam sam 272K Mar  6 00:17 20190305_salmon.err\n",
      "-rw-rw-r-- 1 sam sam 1.6M Mar  6 07:46 salmon.isoform.TPM.not_cross_norm\n",
      "-rw-rw-r-- 1 sam sam 1.9M Mar  6 07:46 salmon.isoform.counts.matrix\n",
      "-rw-rw-r-- 1 sam sam 918K Mar  6 07:46 salmon.gene.TPM.not_cross_norm\n",
      "-rw-rw-r-- 1 sam sam 629K Mar  6 07:46 salmon.gene.counts.matrix\n",
      "-rw-rw-r-- 1 sam sam  528 Mar  6 07:46 salmon.isoform.TPM.not_cross_norm.runTMM.R\n",
      "-rw-rw-r-- 1 sam sam 1.7K Mar  6 07:46 salmon.isoform.TPM.not_cross_norm.TMM_info.txt\n",
      "-rw-rw-r-- 1 sam sam 1.3M Mar  6 07:46 salmon.isoform.TMM.EXPR.matrix\n",
      "-rw-rw-r-- 1 sam sam  522 Mar  6 07:46 salmon.gene.TPM.not_cross_norm.runTMM.R\n",
      "-rw-rw-r-- 1 sam sam 1.7K Mar  6 07:46 salmon.gene.TPM.not_cross_norm.TMM_info.txt\n",
      "-rw-rw-r-- 1 sam sam 688K Mar  6 07:46 salmon.gene.TMM.EXPR.matrix\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-reading file: b_44_01/quant.sf\n",
      "-reading file: b_44_02/quant.sf\n",
      "-reading file: b_44_03/quant.sf\n",
      "-reading file: b_K5_01/quant.sf\n",
      "-reading file: b_K5_02/quant.sf\n",
      "-reading file: b_K5_03/quant.sf\n",
      "-reading file: b_K5_04/quant.sf\n",
      "-reading file: b_K5_05/quant.sf\n",
      "-reading file: b_K5_06/quant.sf\n",
      "-reading file: bleached_44_01/quant.sf\n",
      "-reading file: bleached_44_02/quant.sf\n",
      "-reading file: bleached_44_03/quant.sf\n",
      "-reading file: bleached_K5_01/quant.sf\n",
      "-reading file: bleached_K5_02/quant.sf\n",
      "-reading file: bleached_K5_03/quant.sf\n",
      "-reading file: bleached_K5_04/quant.sf\n",
      "-reading file: bleached_K5_05/quant.sf\n",
      "-reading file: bleached_K5_06/quant.sf\n",
      "-reading file: nb_44_04/quant.sf\n",
      "-reading file: nb_44_05/quant.sf\n",
      "-reading file: nb_44_06/quant.sf\n",
      "-reading file: nb_K5_07/quant.sf\n",
      "-reading file: nb_K5_08/quant.sf\n",
      "-reading file: nbK5_09/quant.sf\n",
      "-reading file: non-bleached_44_01/quant.sf\n",
      "-reading file: non-bleached_44_02/quant.sf\n",
      "-reading file: non-bleached_44_03/quant.sf\n",
      "-reading file: non-bleached_K5_01/quant.sf\n",
      "-reading file: non-bleached_K5_02/quant.sf\n",
      "-reading file: non-bleached_K5_03/quant.sf\n",
      "\n",
      "\n",
      "* Outputting combined matrix.\n",
      "\n",
      "/home/shared/Trinityrnaseq-v2.6.6/util/support_scripts/run_TMM_scale_matrix.pl --matrix salmon.isoform.TPM.not_cross_norm > salmon.isoform.TMM.EXPR.matrixCMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.TPM.not_cross_norm.runTMM.R 1>&2 \n",
      "> library(edgeR)\n",
      "Loading required package: limma\n",
      "> \n",
      "> rnaseqMatrix = read.table(\"salmon.isoform.TPM.not_cross_norm\", header=T, row.names=1, com='', check.names=F)\n",
      "> rnaseqMatrix = as.matrix(rnaseqMatrix)\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=factor(colnames(rnaseqMatrix)))\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study$samples$eff.lib.size = exp_study$samples$lib.size * exp_study$samples$norm.factors\n",
      "> write.table(exp_study$samples, file=\"salmon.isoform.TPM.not_cross_norm.TMM_info.txt\", quote=F, sep=\"\\t\", row.names=F)\n",
      "> \n",
      "/home/shared/Trinityrnaseq-v2.6.6/util/support_scripts/run_TMM_scale_matrix.pl --matrix salmon.gene.TPM.not_cross_norm > salmon.gene.TMM.EXPR.matrixCMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.gene.TPM.not_cross_norm.runTMM.R 1>&2 \n",
      "> library(edgeR)\n",
      "Loading required package: limma\n",
      "> \n",
      "> rnaseqMatrix = read.table(\"salmon.gene.TPM.not_cross_norm\", header=T, row.names=1, com='', check.names=F)\n",
      "> rnaseqMatrix = as.matrix(rnaseqMatrix)\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=factor(colnames(rnaseqMatrix)))\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study$samples$eff.lib.size = exp_study$samples$lib.size * exp_study$samples$norm.factors\n",
      "> write.table(exp_study$samples, file=\"salmon.gene.TPM.not_cross_norm.TMM_info.txt\", quote=F, sep=\"\\t\", row.names=F)\n",
      "> \n",
      "\n",
      "real\t0m4.742s\n",
      "user\t0m4.584s\n",
      "sys\t0m0.124s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ${salmon_out_dir}\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/util/abundance_estimates_to_matrix.pl \\\n",
    "--est_method salmon \\\n",
    "--gene_trans_map ${gene_map} \\\n",
    "--out_prefix salmon \\\n",
    "--name_sample_by_basedir \\\n",
    "b_44_01/quant.sf \\\n",
    "b_44_02/quant.sf \\\n",
    "b_44_03/quant.sf \\\n",
    "b_K5_01/quant.sf \\\n",
    "b_K5_02/quant.sf \\\n",
    "b_K5_03/quant.sf \\\n",
    "b_K5_04/quant.sf \\\n",
    "b_K5_05/quant.sf \\\n",
    "b_K5_06/quant.sf \\\n",
    "bleached_44_01/quant.sf \\\n",
    "bleached_44_02/quant.sf \\\n",
    "bleached_44_03/quant.sf \\\n",
    "bleached_K5_01/quant.sf \\\n",
    "bleached_K5_02/quant.sf \\\n",
    "bleached_K5_03/quant.sf \\\n",
    "bleached_K5_04/quant.sf \\\n",
    "bleached_K5_05/quant.sf \\\n",
    "bleached_K5_06/quant.sf \\\n",
    "nb_44_04/quant.sf \\\n",
    "nb_44_05/quant.sf \\\n",
    "nb_44_06/quant.sf \\\n",
    "nb_K5_07/quant.sf \\\n",
    "nb_K5_08/quant.sf \\\n",
    "nbK5_09/quant.sf \\\n",
    "non-bleached_44_01/quant.sf \\\n",
    "non-bleached_44_02/quant.sf \\\n",
    "non-bleached_44_03/quant.sf \\\n",
    "non-bleached_K5_01/quant.sf \\\n",
    "non-bleached_K5_02/quant.sf \\\n",
    "non-bleached_K5_03/quant.sf\n",
    "\n",
    "ls -ltrh\n",
    "\n",
    "sed '/^Subject:/ s/ / porites_abundances JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b_44_01                    b_44_02      b_44_03      b_K5_01      b_K5_02     b_K5_03      b_K5_04      b_K5_05      b_K5_06     bleached_44_01  bleached_44_02  bleached_44_03  bleached_K5_01  bleached_K5_02  bleached_K5_03  bleached_K5_04  bleached_K5_05  bleached_K5_06  nb_44_04    nb_44_05     nb_44_06    nb_K5_07     nb_K5_08     nbK5_09      non-bleached_44_01  non-bleached_44_02  non-bleached_44_03  non-bleached_K5_01  non-bleached_K5_02  non-bleached_K5_03\n",
      "TRINITY_DN168636_c3_g1_i4  64.158549    38.369239    75.865901    10.244672   38.984870    59.690663    29.124642    7.087800    0.000000        64.158749       38.369085       75.865903       10.244483       38.984874       59.690671       29.124626       7.087794        0.000000    41.358019    93.290760   51.201687    34.027180    23.558258    18.392211           41.358088           93.290763           51.201686           34.027185           23.558424           18.391865\n",
      "TRINITY_DN188859_c0_g2_i1  124.603860   232.630210   105.797808   0.000203    1.006521     1.134381     234.914194   121.695888  0.000000        124.604828      232.628094      105.795161      0.000202        1.006095        1.133724        234.914238      121.698912      0.000000    0.000000     114.331889  151.910869   0.158455     0.000000     0.000021            0.000000            114.331764          151.910868          0.155092            0.000000            0.000021\n",
      "TRINITY_DN197948_c1_g1_i1  2557.298542  2041.585661  74.315275    75.666627   1379.123740  0.000000     1360.742636  673.842652  487.404414      2554.626777     2041.458327     74.377675       75.396738       1379.855346     0.000000        1359.668161     674.545625      487.424806  59.297447    25.827250   1900.127240  1749.440003  0.000000     3511.341245         59.267757           25.845969           1900.688466         1749.912026         0.000000            3510.840358\n",
      "TRINITY_DN168408_c3_g2_i4  805.167158   4609.129614  1748.134189  833.179685  475.382212   1373.897399  0.000005     55.065035   0.000000        805.114799      4608.720150     1747.819041     832.842485      475.526048      1373.506555     0.000006        54.830905       0.000000    3878.694212  304.231904  142.121796   211.247692   2500.125443  723.064545          3878.881927         303.905139          142.574890          211.258374          2500.319314         723.114834\n",
      "TRINITY_DN190676_c0_g1_i7  0.000000     0.143526     0.000776     137.866739  7.502746     0.000000     0.000000     12.821717   13.825364       0.000000        0.143427        0.000775        137.843652      7.502737        0.000000        0.000000        12.809576       13.825424   9.855802     41.208701   0.000000     0.000000     109.000000   11.571994           9.834464            41.208569           0.000000            0.000000            109.000000          11.571802\n",
      "TRINITY_DN187673_c1_g1_i1  427.004012   482.416849   703.691463   640.667327  435.467266   662.864999   577.627882   593.035123  243.191937      427.002865      482.430700      703.732184      640.675857      435.476088      662.867600      577.626387      593.038687      243.193766  515.932611   593.336766  598.443713   509.329542   814.448898   814.851228          515.932288          593.337410          598.442443          509.329636          814.447102          814.848764\n",
      "TRINITY_DN172779_c4_g2_i5  17.000000    0.000000     55.000000    17.000000   0.000000     0.000000     25.568362    5.000000    5.000000        17.000000       0.000000        55.000000       17.000000       0.000000        0.000000        25.568502       5.000000        5.000000    1.477424     28.000000   26.346707    0.000000     48.366513    0.000000            1.477425            28.000000           26.346720           0.000000            48.366478           0.000000\n",
      "TRINITY_DN195718_c1_g1_i3  0.000000     0.000000     0.000000     13.413652   0.000000     0.000000     0.000000     0.000000    0.000000        0.000000        0.000000        0.000000        13.414653       0.000000        0.000000        0.000000        0.000000        0.000000    0.000000     22.333429   0.000000     16.489911    30.581973    65.106869           0.000000            22.323930           0.000000            16.762627           30.397090           65.108291\n",
      "TRINITY_DN193323_c0_g1_i7  0.000000     0.000000     0.000000     40.061557   8.473426     18.282218    0.000000     46.476395   0.000000        0.000000        0.000000        0.000000        40.062420       8.473547        18.286658       0.000000        46.476208       0.000000    0.000000     0.000000    0.000000     0.000000     8.914547     0.000000            0.000000            0.000000            0.000000            0.000000            8.914546            0.000000\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "head ${salmon_out_dir}/salmon.isoform.counts.matrix \\\n",
    "| column -t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "$VAR1 = {\n",
      "          '44' => [\n",
      "                    'b_44_01',\n",
      "                    'b_44_02',\n",
      "                    'b_44_03',\n",
      "                    'nb_44_04',\n",
      "                    'nb_44_05',\n",
      "                    'nb_44_06'\n",
      "                  ],\n",
      "          'non-bleached' => [\n",
      "                              'non-bleached_K5_01',\n",
      "                              'non-bleached_K5_02',\n",
      "                              'non-bleached_K5_03',\n",
      "                              'non-bleached_44_01',\n",
      "                              'non-bleached_44_02',\n",
      "                              'non-bleached_44_03'\n",
      "                            ],\n",
      "          'k5' => [\n",
      "                    'b_K5_01',\n",
      "                    'b_K5_02',\n",
      "                    'b_K5_03',\n",
      "                    'b_K5_04',\n",
      "                    'b_K5_05',\n",
      "                    'b_K5_06',\n",
      "                    'nb_K5_07',\n",
      "                    'nb_K5_08',\n",
      "                    'nbK5_09'\n",
      "                  ],\n",
      "          'bleached' => [\n",
      "                          'bleached_K5_01',\n",
      "                          'bleached_K5_02',\n",
      "                          'bleached_K5_03',\n",
      "                          'bleached_K5_04',\n",
      "                          'bleached_K5_05',\n",
      "                          'bleached_K5_06',\n",
      "                          'bleached_44_01',\n",
      "                          'bleached_44_02',\n",
      "                          'bleached_44_03'\n",
      "                        ]\n",
      "        };\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.44_vs_bleached.44.vs.bleached.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/porites/20190305_salmon_abundance_porites/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(1,2,3,19,20,21,13,14,15,16,17,18,10,11,12)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"44\", 6), rep(\"bleached\", 9)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"44\", \"bleached\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"44\", sampleB=\"bleached\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.44_vs_bleached.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.44_vs_bleached.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.44_vs_bleached.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.44_vs_k5.44.vs.k5.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/porites/20190305_salmon_abundance_porites/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(1,2,3,19,20,21,4,5,6,7,8,9,22,23,24)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"44\", 6), rep(\"k5\", 9)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"44\", \"k5\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"44\", sampleB=\"k5\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.44_vs_k5.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.44_vs_k5.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.44_vs_k5.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.44_vs_non-bleached.44.vs.non-bleached.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/porites/20190305_salmon_abundance_porites/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(1,2,3,19,20,21,28,29,30,25,26,27)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"44\", 6), rep(\"non-bleached\", 6)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"44\", \"non-bleached\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"44\", sampleB=\"non-bleached\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.44_vs_non-bleached.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.44_vs_non-bleached.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.44_vs_non-bleached.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.bleached_vs_k5.bleached.vs.k5.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/porites/20190305_salmon_abundance_porites/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(13,14,15,16,17,18,10,11,12,4,5,6,7,8,9,22,23,24)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"bleached\", 9), rep(\"k5\", 9)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"bleached\", \"k5\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"bleached\", sampleB=\"k5\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.bleached_vs_k5.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.bleached_vs_k5.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.bleached_vs_k5.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.bleached_vs_non-bleached.bleached.vs.non-bleached.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/porites/20190305_salmon_abundance_porites/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(13,14,15,16,17,18,10,11,12,28,29,30,25,26,27)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"bleached\", 9), rep(\"non-bleached\", 6)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"bleached\", \"non-bleached\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"bleached\", sampleB=\"non-bleached\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.k5_vs_non-bleached.k5.vs.non-bleached.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/porites/20190305_salmon_abundance_porites/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(4,5,6,7,8,9,22,23,24,28,29,30,25,26,27)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"k5\", 9), rep(\"non-bleached\", 6)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"k5\", \"non-bleached\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"k5\", sampleB=\"non-bleached\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.k5_vs_non-bleached.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.k5_vs_non-bleached.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.k5_vs_non-bleached.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Got 30 samples, and got: 31 data fields.\n",
      "Header: b_44_01\tb_44_02\tb_44_03\tb_K5_01\tb_K5_02\tb_K5_03\tb_K5_04\tb_K5_05\tb_K5_06\tbleached_44_01\tbleached_44_02\tbleached_44_03\tbleached_K5_01\tbleached_K5_02\tbleached_K5_03\tbleached_K5_04\tbleached_K5_05\tbleached_K5_06\tnb_44_04\tnb_44_05\tnb_44_06\tnb_K5_07\tnb_K5_08\tnbK5_09\tnon-bleached_44_01\tnon-bleached_44_02\tnon-bleached_44_03\tnon-bleached_K5_01\tnon-bleached_K5_02\tnon-bleached_K5_03\n",
      "Next: TRINITY_DN168636_c3_g1_i4\t64.158549\t38.369239\t75.865901\t10.244672\t38.984870\t59.690663\t29.124642\t7.087800\t0.000000\t64.158749\t38.369085\t75.865903\t10.244483\t38.984874\t59.690671\t29.124626\t7.087794\t0.000000\t41.358019\t93.290760\t51.201687\t34.027180\t23.558258\t18.392211\t41.358088\t93.290763\t51.201686\t34.027185\t23.558424\t18.391865\n",
      "\n",
      "$VAR1 = {\n",
      "          'b_K5_04' => 7,\n",
      "          'bleached_K5_02' => 14,\n",
      "          'b_K5_03' => 6,\n",
      "          'non-bleached_44_01' => 25,\n",
      "          'nb_44_05' => 20,\n",
      "          'nb_K5_08' => 23,\n",
      "          'bleached_K5_01' => 13,\n",
      "          'bleached_K5_03' => 15,\n",
      "          'nb_K5_07' => 22,\n",
      "          'non-bleached_K5_03' => 30,\n",
      "          'nbK5_09' => 24,\n",
      "          'b_44_01' => 1,\n",
      "          'bleached_K5_04' => 16,\n",
      "          'nb_44_04' => 19,\n",
      "          'non-bleached_K5_02' => 29,\n",
      "          'bleached_K5_05' => 17,\n",
      "          'b_44_03' => 3,\n",
      "          'nb_44_06' => 21,\n",
      "          'bleached_44_03' => 12,\n",
      "          'bleached_K5_06' => 18,\n",
      "          'b_44_02' => 2,\n",
      "          'non-bleached_K5_01' => 28,\n",
      "          'non-bleached_44_02' => 26,\n",
      "          'b_K5_02' => 5,\n",
      "          'bleached_44_02' => 11,\n",
      "          'b_K5_06' => 9,\n",
      "          'bleached_44_01' => 10,\n",
      "          'non-bleached_44_03' => 27,\n",
      "          'b_K5_01' => 4,\n",
      "          'b_K5_05' => 8\n",
      "        };\n",
      "Contrasts to perform are: $VAR1 = [\n",
      "          [\n",
      "            '44',\n",
      "            'bleached'\n",
      "          ],\n",
      "          [\n",
      "            '44',\n",
      "            'k5'\n",
      "          ],\n",
      "          [\n",
      "            '44',\n",
      "            'non-bleached'\n",
      "          ],\n",
      "          [\n",
      "            'bleached',\n",
      "            'k5'\n",
      "          ],\n",
      "          [\n",
      "            'bleached',\n",
      "            'non-bleached'\n",
      "          ],\n",
      "          [\n",
      "            'k5',\n",
      "            'non-bleached'\n",
      "          ]\n",
      "        ];\n",
      "Loading required package: limma\n",
      "Loading required package: limma\n",
      "Loading required package: limma\n",
      "Loading required package: limma\n",
      "Loading required package: limma\n",
      "Loading required package: limma\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts\n",
    "\n",
    "/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/run_DE_analysis.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.counts.matrix \\\n",
    "--method edgeR \\\n",
    "--samples_file ${samples} \n",
    "\n",
    "sed '/^Subject:/ s/ / porites_edgeR JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 14M\n",
      "-rw-rw-r-- 1 sam sam  13M Jan 30 11:09 Trinity.sp.coral.fasta\n",
      "-rw-rw-r-- 1 sam sam 257K Jan 30 11:09 Trinity.sp.coral.fasta.fai\n",
      "-rw-rw-r-- 1 sam sam 280K Jan 30 21:09 Trinity.sp.coral.fasta.gene_trans_map\n",
      "drwxrwxr-x 2 sam sam 4.0K Jan 30 21:09 Trinity.sp.coral.fasta.salmon_quasi.idx\n",
      "drwxrwxr-x 2 sam sam 4.0K Jan 31 08:00 edgeR.3209.dir\n",
      "drwxrwxr-x 2 sam sam 4.0K Mar  6 08:54 edgeR.24054.dir\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -ltrh /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> library(cluster)\n",
      "> library(Biobase)\n",
      "> library(qvalue)\n",
      "> library(fastcluster)\n",
      "> options(stringsAsFactors = FALSE)\n",
      "> NO_REUSE = F\n",
      "> \n",
      "> # try to reuse earlier-loaded data if possible\n",
      "> if (file.exists(\"diffExpr.P0.001_C2.matrix.RData\") && ! NO_REUSE) {\n",
      "+     print('RESTORING DATA FROM EARLIER ANALYSIS')\n",
      "+     load(\"diffExpr.P0.001_C2.matrix.RData\")\n",
      "+ } else {\n",
      "+     print('Reading matrix file.')\n",
      "+     primary_data = read.table(\"diffExpr.P0.001_C2.matrix\", header=T, com='', row.names=1, check.names=F)\n",
      "+     primary_data = as.matrix(primary_data)\n",
      "+ }\n",
      "[1] \"Reading matrix file.\"\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/heatmap.3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/misc_rnaseq_funcs.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/pairs3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/vioplot2.R\")\n",
      "> data = primary_data\n",
      "> myheatcol = colorpanel(75, 'purple','black','yellow')\n",
      "> samples_data = read.table(\"/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_44_vs_k5_trinity_sample_list.txt\", header=F, check.names=F, fill=T)\n",
      "> samples_data = samples_data[samples_data[,2] != '',]\n",
      "> colnames(samples_data) = c('sample_name', 'replicate_name')\n",
      "> sample_types = as.character(unique(samples_data[,1]))\n",
      "> rep_names = as.character(samples_data[,2])\n",
      "> data = data[, colnames(data) %in% rep_names, drop=F ]\n",
      "> nsamples = length(sample_types)\n",
      "> sample_colors = rainbow(nsamples)\n",
      "> names(sample_colors) = sample_types\n",
      "> sample_type_list = list()\n",
      "> for (i in 1:nsamples) {\n",
      "+     samples_want = samples_data[samples_data[,1]==sample_types[i], 2]\n",
      "+     sample_type_list[[sample_types[i]]] = as.vector(samples_want)\n",
      "+ }\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> initial_matrix = data # store before doing various data transformations\n",
      "> data = log2(data+1)\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> sampleAnnotations = matrix(ncol=ncol(data),nrow=nsamples)\n",
      "> for (i in 1:nsamples) {\n",
      "+   sampleAnnotations[i,] = colnames(data) %in% sample_type_list[[sample_types[i]]]\n",
      "+ }\n",
      "> sampleAnnotations = apply(sampleAnnotations, 1:2, function(x) as.logical(x))\n",
      "> sampleAnnotations = sample_matrix_to_color_assignments(sampleAnnotations, col=sample_colors)\n",
      "> rownames(sampleAnnotations) = as.vector(sample_types)\n",
      "> colnames(sampleAnnotations) = colnames(data)\n",
      "> data = as.matrix(data) # convert to matrix\n",
      "> \n",
      "> # Centering rows\n",
      "> data = t(scale(t(data), scale=F))\n",
      "> \n",
      "> write.table(data, file=\"diffExpr.P0.001_C2.matrix.log2.centered.dat\", quote=F, sep='\t');\n",
      "> if (nrow(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two rows are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> if (ncol(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two columns are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> sample_cor = cor(data, method='pearson', use='pairwise.complete.obs')\n",
      "> write.table(sample_cor, file=\"diffExpr.P0.001_C2.matrix.log2.centered.sample_cor.dat\", quote=F, sep='\t')\n",
      "> sample_dist = dist(t(data), method='euclidean')\n",
      "> hc_samples = hclust(sample_dist, method='complete')\n",
      "> pdf(\"diffExpr.P0.001_C2.matrix.log2.centered.sample_cor_matrix.pdf\")\n",
      "> sample_cor_for_plot = sample_cor\n",
      "> heatmap.3(sample_cor_for_plot, dendrogram='both', Rowv=as.dendrogram(hc_samples), Colv=as.dendrogram(hc_samples), col = myheatcol, scale='none', symm=TRUE, key=TRUE,density.info='none', trace='none', symkey=FALSE, symbreaks=F, margins=c(10,10), cexCol=1, cexRow=1, cex.main=0.75, main=paste(\"sample correlation matrix\n",
      "+ \", \"diffExpr.P0.001_C2.matrix.log2.centered\") , ColSideColors=sampleAnnotations, RowSideColors=t(sampleAnnotations))\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> gene_cor = NULL\n",
      "> gene_dist = dist(data, method='euclidean')\n",
      "> if (nrow(data) <= 1) { message('Too few genes to generate heatmap'); quit(status=0); }\n",
      "> hc_genes = hclust(gene_dist, method='complete')\n",
      "> heatmap_data = data\n",
      "> pdf(\"diffExpr.P0.001_C2.matrix.log2.centered.genes_vs_samples_heatmap.pdf\")\n",
      "> heatmap.3(heatmap_data, dendrogram='both', Rowv=as.dendrogram(hc_genes), Colv=as.dendrogram(hc_samples), col=myheatcol, scale=\"none\", density.info=\"none\", trace=\"none\", key=TRUE, keysize=1.2, cexCol=1, margins=c(10,10), cex.main=0.75, main=paste(\"samples vs. features\n",
      "+ \", \"diffExpr.P0.001_C2.matrix.log2.centered\" ) , ColSideColors=sampleAnnotations)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> save(list=ls(all=TRUE), file=\"diffExpr.P0.001_C2.matrix.RData\")\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "** Found 5 features as differentially expressed.\n",
      "\n",
      "CMD: /home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/PtR -m diffExpr.P0.001_C2.matrix --log2 --heatmap --min_colSums 0 --min_rowSums 0 --gene_dist euclidean --sample_dist euclidean --sample_cor_matrix --center_rows --save  -s /home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_44_vs_k5_trinity_sample_list.txt\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < diffExpr.P0.001_C2.matrix.R\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘fastcluster’\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    hclust\n",
      "\n",
      "for plotting:: min.raw: -0.945154412097032 max.raw: 1\n",
      "for plotting:: min.raw: -9.64888699134391 max.raw: 9.64888699134391\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.24054.dir/\n",
    "\n",
    "/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/analyze_diff_expr.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.TMM.EXPR.matrix \\\n",
    "--samples ${samples}\n",
    "\n",
    "sed '/^Subject:/ s/ / porites_edgeR JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> library(cluster)\n",
      "> library(Biobase)\n",
      "> library(qvalue)\n",
      "> library(fastcluster)\n",
      "> options(stringsAsFactors = FALSE)\n",
      "> NO_REUSE = F\n",
      "> \n",
      "> # try to reuse earlier-loaded data if possible\n",
      "> if (file.exists(\"diffExpr.P0.05_C1.matrix.RData\") && ! NO_REUSE) {\n",
      "+     print('RESTORING DATA FROM EARLIER ANALYSIS')\n",
      "+     load(\"diffExpr.P0.05_C1.matrix.RData\")\n",
      "+ } else {\n",
      "+     print('Reading matrix file.')\n",
      "+     primary_data = read.table(\"diffExpr.P0.05_C1.matrix\", header=T, com='', row.names=1, check.names=F)\n",
      "+     primary_data = as.matrix(primary_data)\n",
      "+ }\n",
      "[1] \"Reading matrix file.\"\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/heatmap.3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/misc_rnaseq_funcs.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/pairs3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/vioplot2.R\")\n",
      "> data = primary_data\n",
      "> myheatcol = colorpanel(75, 'purple','black','yellow')\n",
      "> samples_data = read.table(\"/home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_44_vs_k5_trinity_sample_list.txt\", header=F, check.names=F, fill=T)\n",
      "> samples_data = samples_data[samples_data[,2] != '',]\n",
      "> colnames(samples_data) = c('sample_name', 'replicate_name')\n",
      "> sample_types = as.character(unique(samples_data[,1]))\n",
      "> rep_names = as.character(samples_data[,2])\n",
      "> data = data[, colnames(data) %in% rep_names, drop=F ]\n",
      "> nsamples = length(sample_types)\n",
      "> sample_colors = rainbow(nsamples)\n",
      "> names(sample_colors) = sample_types\n",
      "> sample_type_list = list()\n",
      "> for (i in 1:nsamples) {\n",
      "+     samples_want = samples_data[samples_data[,1]==sample_types[i], 2]\n",
      "+     sample_type_list[[sample_types[i]]] = as.vector(samples_want)\n",
      "+ }\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> initial_matrix = data # store before doing various data transformations\n",
      "> data = log2(data+1)\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> sampleAnnotations = matrix(ncol=ncol(data),nrow=nsamples)\n",
      "> for (i in 1:nsamples) {\n",
      "+   sampleAnnotations[i,] = colnames(data) %in% sample_type_list[[sample_types[i]]]\n",
      "+ }\n",
      "> sampleAnnotations = apply(sampleAnnotations, 1:2, function(x) as.logical(x))\n",
      "> sampleAnnotations = sample_matrix_to_color_assignments(sampleAnnotations, col=sample_colors)\n",
      "> rownames(sampleAnnotations) = as.vector(sample_types)\n",
      "> colnames(sampleAnnotations) = colnames(data)\n",
      "> data = as.matrix(data) # convert to matrix\n",
      "> \n",
      "> # Centering rows\n",
      "> data = t(scale(t(data), scale=F))\n",
      "> \n",
      "> write.table(data, file=\"diffExpr.P0.05_C1.matrix.log2.centered.dat\", quote=F, sep='\t');\n",
      "> if (nrow(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two rows are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> if (ncol(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two columns are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> sample_cor = cor(data, method='pearson', use='pairwise.complete.obs')\n",
      "> write.table(sample_cor, file=\"diffExpr.P0.05_C1.matrix.log2.centered.sample_cor.dat\", quote=F, sep='\t')\n",
      "> sample_dist = dist(t(data), method='euclidean')\n",
      "> hc_samples = hclust(sample_dist, method='complete')\n",
      "> pdf(\"diffExpr.P0.05_C1.matrix.log2.centered.sample_cor_matrix.pdf\")\n",
      "> sample_cor_for_plot = sample_cor\n",
      "> heatmap.3(sample_cor_for_plot, dendrogram='both', Rowv=as.dendrogram(hc_samples), Colv=as.dendrogram(hc_samples), col = myheatcol, scale='none', symm=TRUE, key=TRUE,density.info='none', trace='none', symkey=FALSE, symbreaks=F, margins=c(10,10), cexCol=1, cexRow=1, cex.main=0.75, main=paste(\"sample correlation matrix\n",
      "+ \", \"diffExpr.P0.05_C1.matrix.log2.centered\") , ColSideColors=sampleAnnotations, RowSideColors=t(sampleAnnotations))\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> gene_cor = NULL\n",
      "> gene_dist = dist(data, method='euclidean')\n",
      "> if (nrow(data) <= 1) { message('Too few genes to generate heatmap'); quit(status=0); }\n",
      "> hc_genes = hclust(gene_dist, method='complete')\n",
      "> heatmap_data = data\n",
      "> pdf(\"diffExpr.P0.05_C1.matrix.log2.centered.genes_vs_samples_heatmap.pdf\")\n",
      "> heatmap.3(heatmap_data, dendrogram='both', Rowv=as.dendrogram(hc_genes), Colv=as.dendrogram(hc_samples), col=myheatcol, scale=\"none\", density.info=\"none\", trace=\"none\", key=TRUE, keysize=1.2, cexCol=1, margins=c(10,10), cex.main=0.75, main=paste(\"samples vs. features\n",
      "+ \", \"diffExpr.P0.05_C1.matrix.log2.centered\" ) , ColSideColors=sampleAnnotations)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> save(list=ls(all=TRUE), file=\"diffExpr.P0.05_C1.matrix.RData\")\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "** Found 33 features as differentially expressed.\n",
      "\n",
      "CMD: /home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/PtR -m diffExpr.P0.05_C1.matrix --log2 --heatmap --min_colSums 0 --min_rowSums 0 --gene_dist euclidean --sample_dist euclidean --sample_cor_matrix --center_rows --save  -s /home/sam/gitrepos/coral_rnaseq_2017/scripts/porites_b_vs_nb_44_vs_k5_trinity_sample_list.txt\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < diffExpr.P0.05_C1.matrix.R\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘fastcluster’\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    hclust\n",
      "\n",
      "for plotting:: min.raw: -0.477775106493466 max.raw: 1\n",
      "for plotting:: min.raw: -10.7092621731909 max.raw: 10.7092621731909\n",
      "\n",
      "real\t0m1.875s\n",
      "user\t0m1.788s\n",
      "sys\t0m0.036s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/porites/20190129_sp_coral_transcripts/edgeR.24054.dir/\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/analyze_diff_expr.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.TMM.EXPR.matrix \\\n",
    "--samples ${samples} \\\n",
    "-C 1 \\\n",
    "-P 0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
