{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TODAY'S DATE\n",
      "Tue Mar  5 22:52:40 PST 2019\n",
      "------------\n",
      "\n",
      "Distributor ID:\tUbuntu\n",
      "Description:\tUbuntu 16.04.5 LTS\n",
      "Release:\t16.04\n",
      "Codename:\txenial\n",
      "\n",
      "------------\n",
      "HOSTNAME: \n",
      "jpg-d20\n",
      "\n",
      "------------\n",
      "Computer Specs:\n",
      "\n",
      "Architecture:          x86_64\n",
      "CPU op-mode(s):        32-bit, 64-bit\n",
      "Byte Order:            Little Endian\n",
      "CPU(s):                24\n",
      "On-line CPU(s) list:   0-23\n",
      "Thread(s) per core:    2\n",
      "Core(s) per socket:    6\n",
      "Socket(s):             2\n",
      "NUMA node(s):          2\n",
      "Vendor ID:             GenuineIntel\n",
      "CPU family:            6\n",
      "Model:                 44\n",
      "Model name:            Intel(R) Xeon(R) CPU           X5680  @ 3.33GHz\n",
      "Stepping:              2\n",
      "CPU MHz:               1596.000\n",
      "CPU max MHz:           3326.0000\n",
      "CPU min MHz:           1596.0000\n",
      "BogoMIPS:              6666.90\n",
      "Virtualization:        VT-x\n",
      "L1d cache:             32K\n",
      "L1i cache:             32K\n",
      "L2 cache:              256K\n",
      "L3 cache:              12288K\n",
      "NUMA node0 CPU(s):     0,2,4,6,8,10,12,14,16,18,20,22\n",
      "NUMA node1 CPU(s):     1,3,5,7,9,11,13,15,17,19,21,23\n",
      "Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 popcnt aes lahf_lm epb kaiser tpr_shadow vnmi flexpriority ept vpid dtherm ida arat\n",
      "\n",
      "------------\n",
      "\n",
      "Memory Specs\n",
      "\n",
      "              total        used        free      shared  buff/cache   available\n",
      "Mem:            94G        1.8G        3.5G        1.0G         89G         90G\n",
      "Swap:          128G          0B        128G\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No LSB modules are available.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"TODAY'S DATE\"\n",
    "date\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "lsb_release -a\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"HOSTNAME: \"\n",
    "hostname\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"Computer Specs:\"\n",
    "echo \"\"\n",
    "lscpu\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "echo \"Memory Specs\"\n",
    "echo \"\"\n",
    "free -mh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: salmon_out_dir=/media/sam/4TB_toshiba/montipora/20190305_salmon_abundance_montipora\n",
      "env: coral_transcriptome=/media/sam/4TB_toshiba/montipora/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta\n",
      "env: coral_fasta_index=/media/sam/4TB_toshiba/montipora/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.fai\n",
      "env: samples=/home/sam/gitrepos/coral_rnaseq_2017/scripts/montipora_b_vs_nb_44_vs_k4_trinity_sample_list.txt\n",
      "env: gene_map=/media/sam/4TB_toshiba/montipora/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.gene_trans_map\n",
      "env: salmon_stdout=20190305_salmon.out\n",
      "env: salmon_stderr=20190305_salmon.err\n"
     ]
    }
   ],
   "source": [
    "%env salmon_out_dir=/media/sam/4TB_toshiba/montipora/20190305_salmon_abundance_montipora\n",
    "%env coral_transcriptome=/media/sam/4TB_toshiba/montipora/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta\n",
    "%env coral_fasta_index=/media/sam/4TB_toshiba/montipora/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.fai\n",
    "%env samples=/home/sam/gitrepos/coral_rnaseq_2017/scripts/montipora_b_vs_nb_44_vs_k4_trinity_sample_list.txt\n",
    "%env gene_map=/media/sam/4TB_toshiba/montipora/20190129_sp_coral_transcripts/Trinity.sp.coral.fasta.gene_trans_map\n",
    "\n",
    "%env salmon_stdout=20190305_salmon.out\n",
    "%env salmon_stderr=20190305_salmon.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "44  b_44_01 4R041-L6-P10-GTCCGC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P10-GTCCGC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "44  b_44_02 4R041-L6-P03-GTTTCG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P03-GTTTCG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "44  b_44_03 4R041-L6-P04-ATCACG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P04-ATCACG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "44  nb_44_04 4R041-L6-P11-GAGTGG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P11-GAGTGG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "44  nb_44_05 4R041-L6-P05-TAGCTT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P05-TAGCTT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "44  nb_44_06 4R041-L6-P12-GCCAAT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P12-GCCAAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k4  b_k4_01 4R041-L6-P09-GGCTAC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P09-GGCTAC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k4  b_k4_02 4R041-L6-P06-AGTTCC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P06-AGTTCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k4  b_k4_03 4R041-L6-P07-ATGTCA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P07-ATGTCA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k4  nb_k4_04 4R041-L6-P08-ACTGAT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P08-ACTGAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k4  nb_k4_05 4R041-L6-P01-AGTCAA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P01-AGTCAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "k4  nb_k4_06 4R041-L6-P02-CTTGTA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P02-CTTGTA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_k4_01 4R041-L6-P09-GGCTAC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P09-GGCTAC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_44_01 4R041-L6-P10-GTCCGC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P10-GTCCGC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_k4_02 4R041-L6-P06-AGTTCC-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P06-AGTTCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_44_02 4R041-L6-P03-GTTTCG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P03-GTTTCG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_44_03 4R041-L6-P04-ATCACG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P04-ATCACG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "bleached  bleached_k4_03 4R041-L6-P07-ATGTCA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P07-ATGTCA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached non-bleached_k4_01  4R041-L6-P08-ACTGAT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P08-ACTGAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_k4_02 4R041-L6-P01-AGTCAA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P01-AGTCAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_44_01 4R041-L6-P11-GAGTGG-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P11-GAGTGG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_k4_03 4R041-L6-P02-CTTGTA-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P02-CTTGTA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_44_02 4R041-L6-P05-TAGCTT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P05-TAGCTT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "non-bleached  non-bleached_44_03 4R041-L6-P12-GCCAAT-READ1-Sequences.txt.gz_val_1.fq.gz  4R041-L6-P12-GCCAAT-READ2-Sequences.txt.gz_val_2.fq.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat ${samples}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "real\t68m48.008s\n",
      "user\t921m1.676s\n",
      "sys\t5m13.596s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /mnt/data/coral_RNAseq_2017/montipora/20180311_fastqc_trimming/trimmed\n",
    "\n",
    "time /home/shared/Trinityrnaseq-v2.6.6/util/align_and_estimate_abundance.pl \\\n",
    "--output_dir ${salmon_out_dir} \\\n",
    "--transcripts ${coral_transcriptome} \\\n",
    "--seqType fq \\\n",
    "--samples_file ${samples} \\\n",
    "--SS_lib_type RF \\\n",
    "--est_method salmon \\\n",
    "--aln_method bowtie2 \\\n",
    "--trinity_mode \\\n",
    "--prep_reference \\\n",
    "--thread_count 23 \\\n",
    "1> ${salmon_out_dir}/${salmon_stdout} \\\n",
    "2> ${salmon_out_dir}/${salmon_stderr}\n",
    "sed '/^Subject:/ s/ / montipora_salmon JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 41G\n",
      "-rw-rw-r-- 1 sam sam 4.1K Apr 13  2018 4R041-L6-P01-AGTCAA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 1.4G Apr 13  2018 4R041-L6-P01-AGTCAA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 1.3G Apr 13  2018 4R041-L6-P01-AGTCAA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.4K Apr 13  2018 4R041-L6-P01-AGTCAA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 4.2K Apr 13  2018 4R041-L6-P02-CTTGTA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 1.6G Apr 13  2018 4R041-L6-P02-CTTGTA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 1.9G Apr 13  2018 4R041-L6-P02-CTTGTA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.4K Apr 13  2018 4R041-L6-P02-CTTGTA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 4.1K Apr 13  2018 4R041-L6-P03-GTTTCG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 1.2G Apr 13  2018 4R041-L6-P03-GTTTCG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 1.4G Apr 13  2018 4R041-L6-P03-GTTTCG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.4K Apr 13  2018 4R041-L6-P03-GTTTCG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 4.2K Apr 13  2018 4R041-L6-P04-ATCACG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 1.3G Apr 13  2018 4R041-L6-P04-ATCACG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 1.4G Apr 13  2018 4R041-L6-P04-ATCACG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.4K Apr 13  2018 4R041-L6-P04-ATCACG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 4.1K Apr 13  2018 4R041-L6-P05-TAGCTT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 1.7G Apr 13  2018 4R041-L6-P05-TAGCTT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 2.0G Apr 13  2018 4R041-L6-P05-TAGCTT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.4K Apr 13  2018 4R041-L6-P05-TAGCTT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 4.2K Apr 13  2018 4R041-L6-P06-AGTTCC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 2.0G Apr 13  2018 4R041-L6-P06-AGTTCC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 2.3G Apr 13  2018 4R041-L6-P06-AGTTCC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.4K Apr 13  2018 4R041-L6-P06-AGTTCC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 4.2K Apr 13  2018 4R041-L6-P07-ATGTCA-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 1.5G Apr 13  2018 4R041-L6-P07-ATGTCA-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 1.6G Apr 13  2018 4R041-L6-P07-ATGTCA-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.4K Apr 13  2018 4R041-L6-P07-ATGTCA-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 4.2K Apr 13  2018 4R041-L6-P08-ACTGAT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 2.3G Apr 13  2018 4R041-L6-P08-ACTGAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 2.1G Apr 13  2018 4R041-L6-P08-ACTGAT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.4K Apr 13  2018 4R041-L6-P08-ACTGAT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 4.2K Apr 13  2018 4R041-L6-P09-GGCTAC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 1.7G Apr 13  2018 4R041-L6-P09-GGCTAC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 1.9G Apr 13  2018 4R041-L6-P09-GGCTAC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.4K Apr 13  2018 4R041-L6-P09-GGCTAC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 4.1K Apr 13  2018 4R041-L6-P10-GTCCGC-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 778M Apr 13  2018 4R041-L6-P10-GTCCGC-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 897M Apr 13  2018 4R041-L6-P10-GTCCGC-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.3K Apr 13  2018 4R041-L6-P10-GTCCGC-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 4.3K Apr 13  2018 4R041-L6-P11-GAGTGG-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 2.1G Apr 13  2018 4R041-L6-P11-GAGTGG-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 2.4G Apr 13  2018 4R041-L6-P11-GAGTGG-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.5K Apr 13  2018 4R041-L6-P11-GAGTGG-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 4.2K Apr 13  2018 4R041-L6-P12-GCCAAT-READ1-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 2.1G Apr 13  2018 4R041-L6-P12-GCCAAT-READ1-Sequences.txt.gz_val_1.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 2.4G Apr 13  2018 4R041-L6-P12-GCCAAT-READ2-Sequences.txt.gz_val_2.fq.gz\n",
      "-rw-rw-r-- 1 sam sam 4.4K Apr 13  2018 4R041-L6-P12-GCCAAT-READ2-Sequences.txt.gz_trimming_report.txt\n",
      "-rw-rw-r-- 1 sam sam 177K Apr 15  2018 stderr.log\n",
      "drwxrwxr-x 4 sam sam 4.0K Apr 15  2018 fastqc\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:06 b_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:07 b_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:09 b_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:13 nb_44_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:16 nb_44_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:21 nb_44_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:24 b_k4_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:27 b_k4_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:30 b_k4_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:33 nb_k4_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:36 nb_k4_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:38 nb_k4_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:41 bleached_k4_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:43 bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:47 bleached_k4_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:49 bleached_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:51 bleached_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:54 bleached_k4_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:58 non-bleached_k4_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:00 non-bleached_k4_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:04 non-bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:07 non-bleached_k4_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:10 non-bleached_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:14 non-bleached_44_03\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -ltrh /mnt/data/coral_RNAseq_2017/montipora/20180311_fastqc_trimming/trimmed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 352K\n",
      "-rw-rw-r-- 1 sam sam    0 Mar  5 23:05 20190305_salmon.out\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:06 b_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:07 b_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:09 b_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:13 nb_44_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:16 nb_44_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:21 nb_44_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:24 b_k4_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:27 b_k4_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:30 b_k4_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:33 nb_k4_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:36 nb_k4_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:38 nb_k4_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:41 bleached_k4_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:43 bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:47 bleached_k4_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:49 bleached_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:51 bleached_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:54 bleached_k4_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:58 non-bleached_k4_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:00 non-bleached_k4_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:04 non-bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:07 non-bleached_k4_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:10 non-bleached_44_02\n",
      "-rw-rw-r-- 1 sam sam 249K Mar  6 00:14 20190305_salmon.err\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:14 non-bleached_44_03\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mv /mnt/data/coral_RNAseq_2017/montipora/20180311_fastqc_trimming/trimmed/[bn]* \\\n",
    "${salmon_out_dir}\n",
    "\n",
    "ls -ltrh ${salmon_out_dir}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 6.9M\n",
      "-rw-rw-r-- 1 sam sam    0 Mar  5 23:05 20190305_salmon.out\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:06 b_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:07 b_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:09 b_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:13 nb_44_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:16 nb_44_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:21 nb_44_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:24 b_k4_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:27 b_k4_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:30 b_k4_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:33 nb_k4_04\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:36 nb_k4_05\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:38 nb_k4_06\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:41 bleached_k4_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:43 bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:47 bleached_k4_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:49 bleached_44_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:51 bleached_44_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:54 bleached_k4_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  5 23:58 non-bleached_k4_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:00 non-bleached_k4_02\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:04 non-bleached_44_01\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:07 non-bleached_k4_03\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:10 non-bleached_44_02\n",
      "-rw-rw-r-- 1 sam sam 249K Mar  6 00:14 20190305_salmon.err\n",
      "drwxrwxr-x 5 sam sam 4.0K Mar  6 00:14 non-bleached_44_03\n",
      "-rw-rw-r-- 1 sam sam 1.5M Mar  6 07:45 salmon.isoform.TPM.not_cross_norm\n",
      "-rw-rw-r-- 1 sam sam 1.8M Mar  6 07:45 salmon.isoform.counts.matrix\n",
      "-rw-rw-r-- 1 sam sam 792K Mar  6 07:45 salmon.gene.TPM.not_cross_norm\n",
      "-rw-rw-r-- 1 sam sam 579K Mar  6 07:45 salmon.gene.counts.matrix\n",
      "-rw-rw-r-- 1 sam sam  528 Mar  6 07:45 salmon.isoform.TPM.not_cross_norm.runTMM.R\n",
      "-rw-rw-r-- 1 sam sam 1.4K Mar  6 07:45 salmon.isoform.TPM.not_cross_norm.TMM_info.txt\n",
      "-rw-rw-r-- 1 sam sam 1.3M Mar  6 07:45 salmon.isoform.TMM.EXPR.matrix\n",
      "-rw-rw-r-- 1 sam sam  522 Mar  6 07:45 salmon.gene.TPM.not_cross_norm.runTMM.R\n",
      "-rw-rw-r-- 1 sam sam 1.4K Mar  6 07:45 salmon.gene.TPM.not_cross_norm.TMM_info.txt\n",
      "-rw-rw-r-- 1 sam sam 618K Mar  6 07:45 salmon.gene.TMM.EXPR.matrix\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-reading file: b_44_01/quant.sf\n",
      "-reading file: b_44_02/quant.sf\n",
      "-reading file: b_44_03/quant.sf\n",
      "-reading file: b_k4_01/quant.sf\n",
      "-reading file: b_k4_02/quant.sf\n",
      "-reading file: b_k4_03/quant.sf\n",
      "-reading file: bleached_44_01/quant.sf\n",
      "-reading file: bleached_44_02/quant.sf\n",
      "-reading file: bleached_44_03/quant.sf\n",
      "-reading file: bleached_k4_01/quant.sf\n",
      "-reading file: bleached_k4_02/quant.sf\n",
      "-reading file: bleached_k4_03/quant.sf\n",
      "-reading file: nb_44_04/quant.sf\n",
      "-reading file: nb_44_05/quant.sf\n",
      "-reading file: nb_44_06/quant.sf\n",
      "-reading file: nb_k4_04/quant.sf\n",
      "-reading file: nb_k4_05/quant.sf\n",
      "-reading file: nb_k4_06/quant.sf\n",
      "-reading file: non-bleached_44_01/quant.sf\n",
      "-reading file: non-bleached_44_02/quant.sf\n",
      "-reading file: non-bleached_44_03/quant.sf\n",
      "-reading file: non-bleached_k4_01/quant.sf\n",
      "-reading file: non-bleached_k4_02/quant.sf\n",
      "-reading file: non-bleached_k4_03/quant.sf\n",
      "\n",
      "\n",
      "* Outputting combined matrix.\n",
      "\n",
      "/home/shared/Trinityrnaseq-v2.6.6/util/support_scripts/run_TMM_scale_matrix.pl --matrix salmon.isoform.TPM.not_cross_norm > salmon.isoform.TMM.EXPR.matrixCMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.TPM.not_cross_norm.runTMM.R 1>&2 \n",
      "> library(edgeR)\n",
      "Loading required package: limma\n",
      "> \n",
      "> rnaseqMatrix = read.table(\"salmon.isoform.TPM.not_cross_norm\", header=T, row.names=1, com='', check.names=F)\n",
      "> rnaseqMatrix = as.matrix(rnaseqMatrix)\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=factor(colnames(rnaseqMatrix)))\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study$samples$eff.lib.size = exp_study$samples$lib.size * exp_study$samples$norm.factors\n",
      "> write.table(exp_study$samples, file=\"salmon.isoform.TPM.not_cross_norm.TMM_info.txt\", quote=F, sep=\"\\t\", row.names=F)\n",
      "> \n",
      "/home/shared/Trinityrnaseq-v2.6.6/util/support_scripts/run_TMM_scale_matrix.pl --matrix salmon.gene.TPM.not_cross_norm > salmon.gene.TMM.EXPR.matrixCMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.gene.TPM.not_cross_norm.runTMM.R 1>&2 \n",
      "> library(edgeR)\n",
      "Loading required package: limma\n",
      "> \n",
      "> rnaseqMatrix = read.table(\"salmon.gene.TPM.not_cross_norm\", header=T, row.names=1, com='', check.names=F)\n",
      "> rnaseqMatrix = as.matrix(rnaseqMatrix)\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=factor(colnames(rnaseqMatrix)))\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study$samples$eff.lib.size = exp_study$samples$lib.size * exp_study$samples$norm.factors\n",
      "> write.table(exp_study$samples, file=\"salmon.gene.TPM.not_cross_norm.TMM_info.txt\", quote=F, sep=\"\\t\", row.names=F)\n",
      "> \n",
      "\n",
      "real\t0m4.412s\n",
      "user\t0m4.224s\n",
      "sys\t0m0.152s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ${salmon_out_dir}\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/util/abundance_estimates_to_matrix.pl \\\n",
    "--est_method salmon \\\n",
    "--gene_trans_map ${gene_map} \\\n",
    "--out_prefix salmon \\\n",
    "--name_sample_by_basedir \\\n",
    "b_44_01/quant.sf \\\n",
    "b_44_02/quant.sf \\\n",
    "b_44_03/quant.sf \\\n",
    "b_k4_01/quant.sf \\\n",
    "b_k4_02/quant.sf \\\n",
    "b_k4_03/quant.sf \\\n",
    "bleached_44_01/quant.sf \\\n",
    "bleached_44_02/quant.sf \\\n",
    "bleached_44_03/quant.sf \\\n",
    "bleached_k4_01/quant.sf \\\n",
    "bleached_k4_02/quant.sf \\\n",
    "bleached_k4_03/quant.sf \\\n",
    "nb_44_04/quant.sf \\\n",
    "nb_44_05/quant.sf \\\n",
    "nb_44_06/quant.sf \\\n",
    "nb_k4_04/quant.sf \\\n",
    "nb_k4_05/quant.sf \\\n",
    "nb_k4_06/quant.sf \\\n",
    "non-bleached_44_01/quant.sf \\\n",
    "non-bleached_44_02/quant.sf \\\n",
    "non-bleached_44_03/quant.sf \\\n",
    "non-bleached_k4_01/quant.sf \\\n",
    "non-bleached_k4_02/quant.sf \\\n",
    "non-bleached_k4_03/quant.sf\n",
    "\n",
    "ls -ltrh\n",
    "\n",
    "sed '/^Subject:/ s/ / montipora_abundances JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b_44_01                     b_44_02     b_44_03     b_k4_01     b_k4_02     b_k4_03      bleached_44_01  bleached_44_02  bleached_44_03  bleached_k4_01  bleached_k4_02  bleached_k4_03  nb_44_04    nb_44_05     nb_44_06     nb_k4_04     nb_k4_05    nb_k4_06    non-bleached_44_01  non-bleached_44_02  non-bleached_44_03  non-bleached_k4_01  non-bleached_k4_02  non-bleached_k4_03\n",
      "TRINITY_DN174926_c0_g1_i1   36.259871   32.533832   30.332822   46.886252   36.109830    5.558688        36.259324       32.534207       30.333081       46.885564       36.110275       5.558694    46.506661    75.210753    88.031794    64.523261   44.140944   43.641941           46.507002           75.210202           88.031847           64.522609           44.140035           43.642468\n",
      "TRINITY_DN172414_c1_g1_i5   16.000000   173.526638  38.358529   6.062271    34.000000    10.976414       16.000000       173.528280      38.359631       6.062119        34.000000       10.976422   29.701520    10.431028    78.516357    23.957563   14.189704   35.000000           29.701746           10.430817           78.516449           23.956714           14.189620           35.000000\n",
      "TRINITY_DN189559_c0_g1_i1   501.674601  571.341631  557.387187  630.371002  1152.882682  715.928558      501.675083      571.332438      557.372986      630.280496      1152.812808     716.013283  1593.345563  1609.727375  1046.841209  884.467671  606.572029  841.584358          1593.390962         1609.726686         1046.914236         884.491528          606.519851          841.584647\n",
      "TRINITY_DN176445_c3_g3_i10  56.034503   20.794658   40.069362   60.475033   55.838458    54.850574       56.063363       20.787518       40.033928       60.539203       56.184923       54.712269   91.867866    269.439349   88.723362    27.164858   126.047423  154.331727          91.974905           269.423036          88.706915           26.007595           126.055070          154.333027\n",
      "TRINITY_DN188855_c2_g4_i9   215.578260  351.577101  0.000000    276.560682  0.000000     0.000000        215.575907      351.580231      0.000000        276.559935      0.000000        0.000000    63.585033    225.007640   538.701263   631.905575  0.000000    0.000000            63.586648           224.999865          538.700367          631.882790          0.000000            0.000000\n",
      "TRINITY_DN179127_c10_g2_i4  93.566381   98.369601   225.789762  250.529055  307.029437   220.017791      93.604971       98.237436       225.790426      250.573848      307.200063      220.038832  358.866977   278.523564   442.680355   324.878029  225.845232  303.167472          358.827501          278.590517          442.578615          324.828073          225.835370          303.145525\n",
      "TRINITY_DN190946_c1_g2_i9   61.467872   167.122798  0.000000    28.060497   750.052758   856.000000      59.797377       167.637899      0.000000        17.406948       747.666352      856.000000  485.000000   140.254907   268.000000   752.783602  440.000000  1259.000000         485.000000          136.839786          268.000000          747.779010          440.000000          1259.000000\n",
      "TRINITY_DN133735_c0_g1_i1   0.000000    0.000000    0.000000    0.000000    1.000000     33.000000       0.000000        0.000000        0.000000        0.000000        1.000000        33.000000   95.000000    91.000000    0.000000     0.000000    10.000000   66.000000           95.000000           91.000000           0.000000            0.000000            10.000000           66.000000\n",
      "TRINITY_DN179481_c1_g1_i2   0.000000    0.000000    0.000000    0.000000    0.000000     0.000000        0.000000        0.000000        0.000000        0.000000        0.000000        0.000000    0.000000     0.000000     15.936514    0.000000    0.000000    0.000000            0.000000            0.000000            15.993344           0.000000            0.000000            0.000000\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "head ${salmon_out_dir}/salmon.isoform.counts.matrix \\\n",
    "| column -t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "$VAR1 = {\n",
      "          '44' => [\n",
      "                    'b_44_01',\n",
      "                    'b_44_02',\n",
      "                    'b_44_03',\n",
      "                    'nb_44_04',\n",
      "                    'nb_44_05',\n",
      "                    'nb_44_06'\n",
      "                  ],\n",
      "          'k4' => [\n",
      "                    'b_k4_01',\n",
      "                    'b_k4_02',\n",
      "                    'b_k4_03',\n",
      "                    'nb_k4_04',\n",
      "                    'nb_k4_05',\n",
      "                    'nb_k4_06'\n",
      "                  ],\n",
      "          'non-bleached' => [\n",
      "                              'non-bleached_k4_01',\n",
      "                              'non-bleached_k4_02',\n",
      "                              'non-bleached_44_01',\n",
      "                              'non-bleached_k4_03',\n",
      "                              'non-bleached_44_02',\n",
      "                              'non-bleached_44_03'\n",
      "                            ],\n",
      "          'bleached' => [\n",
      "                          'bleached_k4_01',\n",
      "                          'bleached_44_01',\n",
      "                          'bleached_k4_02',\n",
      "                          'bleached_44_02',\n",
      "                          'bleached_44_03',\n",
      "                          'bleached_k4_03'\n",
      "                        ]\n",
      "        };\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.44_vs_bleached.44.vs.bleached.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/montipora/20190305_salmon_abundance_montipora/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(1,2,3,13,14,15,10,7,11,8,9,12)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"44\", 6), rep(\"bleached\", 6)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"44\", \"bleached\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"44\", sampleB=\"bleached\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.44_vs_bleached.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.44_vs_bleached.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.44_vs_bleached.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.44_vs_k4.44.vs.k4.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/montipora/20190305_salmon_abundance_montipora/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(1,2,3,13,14,15,4,5,6,16,17,18)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"44\", 6), rep(\"k4\", 6)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"44\", \"k4\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"44\", sampleB=\"k4\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.44_vs_k4.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.44_vs_k4.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.44_vs_k4.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.44_vs_non-bleached.44.vs.non-bleached.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/montipora/20190305_salmon_abundance_montipora/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(1,2,3,13,14,15,22,23,19,24,20,21)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"44\", 6), rep(\"non-bleached\", 6)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"44\", \"non-bleached\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"44\", sampleB=\"non-bleached\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.44_vs_non-bleached.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.44_vs_non-bleached.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.44_vs_non-bleached.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.bleached_vs_k4.bleached.vs.k4.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/montipora/20190305_salmon_abundance_montipora/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(10,7,11,8,9,12,4,5,6,16,17,18)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"bleached\", 6), rep(\"k4\", 6)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"bleached\", \"k4\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"bleached\", sampleB=\"k4\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.bleached_vs_k4.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.bleached_vs_k4.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.bleached_vs_k4.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.bleached_vs_non-bleached.bleached.vs.non-bleached.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/montipora/20190305_salmon_abundance_montipora/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(10,7,11,8,9,12,22,23,19,24,20,21)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"bleached\", 6), rep(\"non-bleached\", 6)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"bleached\", \"non-bleached\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"bleached\", sampleB=\"non-bleached\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.bleached_vs_non-bleached.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < salmon.isoform.counts.matrix.k4_vs_non-bleached.k4.vs.non-bleached.EdgeR.Rscript\n",
      "> library(edgeR)\n",
      "> \n",
      "> data = read.table(\"/media/sam/4TB_toshiba/montipora/20190305_salmon_abundance_montipora/salmon.isoform.counts.matrix\", header=T, row.names=1, com='')\n",
      "> col_ordering = c(4,5,6,16,17,18,22,23,19,24,20,21)\n",
      "> rnaseqMatrix = data[,col_ordering]\n",
      "> rnaseqMatrix = round(rnaseqMatrix)\n",
      "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
      "> conditions = factor(c(rep(\"k4\", 6), rep(\"non-bleached\", 6)))\n",
      "> \n",
      "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
      "> exp_study = calcNormFactors(exp_study)\n",
      "> exp_study = estimateCommonDisp(exp_study)\n",
      "> exp_study = estimateTagwiseDisp(exp_study)\n",
      "> et = exactTest(exp_study, pair=c(\"k4\", \"non-bleached\"))\n",
      "> tTags = topTags(et,n=NULL)\n",
      "> result_table = tTags$table\n",
      "> result_table = data.frame(sampleA=\"k4\", sampleB=\"non-bleached\", result_table)\n",
      "> result_table$logFC = -1 * result_table$logFC\n",
      "> write.table(result_table, file='salmon.isoform.counts.matrix.k4_vs_non-bleached.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
      "> write.table(rnaseqMatrix, file='salmon.isoform.counts.matrix.k4_vs_non-bleached.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
      "> pdf(\"salmon.isoform.counts.matrix.k4_vs_non-bleached.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
      "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Got 24 samples, and got: 25 data fields.\n",
      "Header: b_44_01\tb_44_02\tb_44_03\tb_k4_01\tb_k4_02\tb_k4_03\tbleached_44_01\tbleached_44_02\tbleached_44_03\tbleached_k4_01\tbleached_k4_02\tbleached_k4_03\tnb_44_04\tnb_44_05\tnb_44_06\tnb_k4_04\tnb_k4_05\tnb_k4_06\tnon-bleached_44_01\tnon-bleached_44_02\tnon-bleached_44_03\tnon-bleached_k4_01\tnon-bleached_k4_02\tnon-bleached_k4_03\n",
      "Next: TRINITY_DN174926_c0_g1_i1\t36.259871\t32.533832\t30.332822\t46.886252\t36.109830\t5.558688\t36.259324\t32.534207\t30.333081\t46.885564\t36.110275\t5.558694\t46.506661\t75.210753\t88.031794\t64.523261\t44.140944\t43.641941\t46.507002\t75.210202\t88.031847\t64.522609\t44.140035\t43.642468\n",
      "\n",
      "$VAR1 = {\n",
      "          'non-bleached_k4_01' => 22,\n",
      "          'bleached_k4_02' => 11,\n",
      "          'b_44_01' => 1,\n",
      "          'b_44_03' => 3,\n",
      "          'bleached_44_01' => 7,\n",
      "          'non-bleached_44_03' => 21,\n",
      "          'non-bleached_44_02' => 20,\n",
      "          'non-bleached_k4_03' => 24,\n",
      "          'b_k4_01' => 4,\n",
      "          'bleached_k4_03' => 12,\n",
      "          'bleached_k4_01' => 10,\n",
      "          'nb_k4_04' => 16,\n",
      "          'b_k4_03' => 6,\n",
      "          'nb_k4_06' => 18,\n",
      "          'bleached_44_03' => 9,\n",
      "          'non-bleached_k4_02' => 23,\n",
      "          'bleached_44_02' => 8,\n",
      "          'nb_44_06' => 15,\n",
      "          'nb_k4_05' => 17,\n",
      "          'b_k4_02' => 5,\n",
      "          'nb_44_05' => 14,\n",
      "          'nb_44_04' => 13,\n",
      "          'b_44_02' => 2,\n",
      "          'non-bleached_44_01' => 19\n",
      "        };\n",
      "Contrasts to perform are: $VAR1 = [\n",
      "          [\n",
      "            '44',\n",
      "            'bleached'\n",
      "          ],\n",
      "          [\n",
      "            '44',\n",
      "            'k4'\n",
      "          ],\n",
      "          [\n",
      "            '44',\n",
      "            'non-bleached'\n",
      "          ],\n",
      "          [\n",
      "            'bleached',\n",
      "            'k4'\n",
      "          ],\n",
      "          [\n",
      "            'bleached',\n",
      "            'non-bleached'\n",
      "          ],\n",
      "          [\n",
      "            'k4',\n",
      "            'non-bleached'\n",
      "          ]\n",
      "        ];\n",
      "Loading required package: limma\n",
      "Loading required package: limma\n",
      "Loading required package: limma\n",
      "Loading required package: limma\n",
      "Loading required package: limma\n",
      "Loading required package: limma\n",
      "\n",
      "real\t0m22.384s\n",
      "user\t0m21.928s\n",
      "sys\t0m0.392s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/montipora/20190129_sp_coral_transcripts\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/run_DE_analysis.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.counts.matrix \\\n",
    "--method edgeR \\\n",
    "--samples_file ${samples} \n",
    "\n",
    "sed '/^Subject:/ s/ / montipora_edgeR JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 17728\n",
      "-rw-rw-r-- 1 sam sam 17491510 Jan 30 11:25 Trinity.sp.coral.fasta\n",
      "-rw-rw-r-- 1 sam sam   308289 Jan 30 11:25 Trinity.sp.coral.fasta.fai\n",
      "-rw-rw-r-- 1 sam sam   334864 Feb  4 14:52 Trinity.sp.coral.fasta.gene_trans_map\n",
      "drwxrwxr-x 2 sam sam     4096 Feb  4 14:52 Trinity.sp.coral.fasta.salmon_quasi.idx\n",
      "drwxrwxr-x 2 sam sam     4096 Feb  7 19:29 edgeR.5710.dir\n",
      "drwxrwxr-x 2 sam sam     4096 Mar  6 07:46 edgeR.22947.dir\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -ltr /media/sam/4TB_toshiba/montipora/20190129_sp_coral_transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> library(cluster)\n",
      "> library(Biobase)\n",
      "> library(qvalue)\n",
      "> library(fastcluster)\n",
      "> options(stringsAsFactors = FALSE)\n",
      "> NO_REUSE = F\n",
      "> \n",
      "> # try to reuse earlier-loaded data if possible\n",
      "> if (file.exists(\"diffExpr.P0.001_C2.matrix.RData\") && ! NO_REUSE) {\n",
      "+     print('RESTORING DATA FROM EARLIER ANALYSIS')\n",
      "+     load(\"diffExpr.P0.001_C2.matrix.RData\")\n",
      "+ } else {\n",
      "+     print('Reading matrix file.')\n",
      "+     primary_data = read.table(\"diffExpr.P0.001_C2.matrix\", header=T, com='', row.names=1, check.names=F)\n",
      "+     primary_data = as.matrix(primary_data)\n",
      "+ }\n",
      "[1] \"Reading matrix file.\"\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/heatmap.3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/misc_rnaseq_funcs.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/pairs3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/vioplot2.R\")\n",
      "> data = primary_data\n",
      "> myheatcol = colorpanel(75, 'purple','black','yellow')\n",
      "> samples_data = read.table(\"/home/sam/gitrepos/coral_rnaseq_2017/scripts/montipora_b_vs_nb_44_vs_k4_trinity_sample_list.txt\", header=F, check.names=F, fill=T)\n",
      "> samples_data = samples_data[samples_data[,2] != '',]\n",
      "> colnames(samples_data) = c('sample_name', 'replicate_name')\n",
      "> sample_types = as.character(unique(samples_data[,1]))\n",
      "> rep_names = as.character(samples_data[,2])\n",
      "> data = data[, colnames(data) %in% rep_names, drop=F ]\n",
      "> nsamples = length(sample_types)\n",
      "> sample_colors = rainbow(nsamples)\n",
      "> names(sample_colors) = sample_types\n",
      "> sample_type_list = list()\n",
      "> for (i in 1:nsamples) {\n",
      "+     samples_want = samples_data[samples_data[,1]==sample_types[i], 2]\n",
      "+     sample_type_list[[sample_types[i]]] = as.vector(samples_want)\n",
      "+ }\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> initial_matrix = data # store before doing various data transformations\n",
      "> data = log2(data+1)\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> sampleAnnotations = matrix(ncol=ncol(data),nrow=nsamples)\n",
      "> for (i in 1:nsamples) {\n",
      "+   sampleAnnotations[i,] = colnames(data) %in% sample_type_list[[sample_types[i]]]\n",
      "+ }\n",
      "> sampleAnnotations = apply(sampleAnnotations, 1:2, function(x) as.logical(x))\n",
      "> sampleAnnotations = sample_matrix_to_color_assignments(sampleAnnotations, col=sample_colors)\n",
      "> rownames(sampleAnnotations) = as.vector(sample_types)\n",
      "> colnames(sampleAnnotations) = colnames(data)\n",
      "> data = as.matrix(data) # convert to matrix\n",
      "> \n",
      "> # Centering rows\n",
      "> data = t(scale(t(data), scale=F))\n",
      "> \n",
      "> write.table(data, file=\"diffExpr.P0.001_C2.matrix.log2.centered.dat\", quote=F, sep='\t');\n",
      "> if (nrow(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two rows are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> if (ncol(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two columns are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> sample_cor = cor(data, method='pearson', use='pairwise.complete.obs')\n",
      "> write.table(sample_cor, file=\"diffExpr.P0.001_C2.matrix.log2.centered.sample_cor.dat\", quote=F, sep='\t')\n",
      "> sample_dist = dist(t(data), method='euclidean')\n",
      "> hc_samples = hclust(sample_dist, method='complete')\n",
      "> pdf(\"diffExpr.P0.001_C2.matrix.log2.centered.sample_cor_matrix.pdf\")\n",
      "> sample_cor_for_plot = sample_cor\n",
      "> heatmap.3(sample_cor_for_plot, dendrogram='both', Rowv=as.dendrogram(hc_samples), Colv=as.dendrogram(hc_samples), col = myheatcol, scale='none', symm=TRUE, key=TRUE,density.info='none', trace='none', symkey=FALSE, symbreaks=F, margins=c(10,10), cexCol=1, cexRow=1, cex.main=0.75, main=paste(\"sample correlation matrix\n",
      "+ \", \"diffExpr.P0.001_C2.matrix.log2.centered\") , ColSideColors=sampleAnnotations, RowSideColors=t(sampleAnnotations))\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> gene_cor = NULL\n",
      "> gene_dist = dist(data, method='euclidean')\n",
      "> if (nrow(data) <= 1) { message('Too few genes to generate heatmap'); quit(status=0); }\n",
      "> hc_genes = hclust(gene_dist, method='complete')\n",
      "> heatmap_data = data\n",
      "> pdf(\"diffExpr.P0.001_C2.matrix.log2.centered.genes_vs_samples_heatmap.pdf\")\n",
      "> heatmap.3(heatmap_data, dendrogram='both', Rowv=as.dendrogram(hc_genes), Colv=as.dendrogram(hc_samples), col=myheatcol, scale=\"none\", density.info=\"none\", trace=\"none\", key=TRUE, keysize=1.2, cexCol=1, margins=c(10,10), cex.main=0.75, main=paste(\"samples vs. features\n",
      "+ \", \"diffExpr.P0.001_C2.matrix.log2.centered\" ) , ColSideColors=sampleAnnotations)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> save(list=ls(all=TRUE), file=\"diffExpr.P0.001_C2.matrix.RData\")\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "** Found 5 features as differentially expressed.\n",
      "\n",
      "CMD: /home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/PtR -m diffExpr.P0.001_C2.matrix --log2 --heatmap --min_colSums 0 --min_rowSums 0 --gene_dist euclidean --sample_dist euclidean --sample_cor_matrix --center_rows --save  -s /home/sam/gitrepos/coral_rnaseq_2017/scripts/montipora_b_vs_nb_44_vs_k4_trinity_sample_list.txt\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < diffExpr.P0.001_C2.matrix.R\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘fastcluster’\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    hclust\n",
      "\n",
      "for plotting:: min.raw: -0.974111205460857 max.raw: 1\n",
      "for plotting:: min.raw: -9.65319575785899 max.raw: 9.65319575785899\n",
      "\n",
      "real\t0m1.836s\n",
      "user\t0m1.732s\n",
      "sys\t0m0.080s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/montipora/20190129_sp_coral_transcripts/edgeR.22947.dir/\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/analyze_diff_expr.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.TMM.EXPR.matrix \\\n",
    "--samples ${samples}\n",
    "\n",
    "sed '/^Subject:/ s/ / montipora_edgeR JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> library(cluster)\n",
      "> library(Biobase)\n",
      "> library(qvalue)\n",
      "> library(fastcluster)\n",
      "> options(stringsAsFactors = FALSE)\n",
      "> NO_REUSE = F\n",
      "> \n",
      "> # try to reuse earlier-loaded data if possible\n",
      "> if (file.exists(\"diffExpr.P0.05_C1.matrix.RData\") && ! NO_REUSE) {\n",
      "+     print('RESTORING DATA FROM EARLIER ANALYSIS')\n",
      "+     load(\"diffExpr.P0.05_C1.matrix.RData\")\n",
      "+ } else {\n",
      "+     print('Reading matrix file.')\n",
      "+     primary_data = read.table(\"diffExpr.P0.05_C1.matrix\", header=T, com='', row.names=1, check.names=F)\n",
      "+     primary_data = as.matrix(primary_data)\n",
      "+ }\n",
      "[1] \"Reading matrix file.\"\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/heatmap.3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/misc_rnaseq_funcs.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/pairs3.R\")\n",
      "> source(\"/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/R/vioplot2.R\")\n",
      "> data = primary_data\n",
      "> myheatcol = colorpanel(75, 'purple','black','yellow')\n",
      "> samples_data = read.table(\"/home/sam/gitrepos/coral_rnaseq_2017/scripts/montipora_b_vs_nb_44_vs_k4_trinity_sample_list.txt\", header=F, check.names=F, fill=T)\n",
      "> samples_data = samples_data[samples_data[,2] != '',]\n",
      "> colnames(samples_data) = c('sample_name', 'replicate_name')\n",
      "> sample_types = as.character(unique(samples_data[,1]))\n",
      "> rep_names = as.character(samples_data[,2])\n",
      "> data = data[, colnames(data) %in% rep_names, drop=F ]\n",
      "> nsamples = length(sample_types)\n",
      "> sample_colors = rainbow(nsamples)\n",
      "> names(sample_colors) = sample_types\n",
      "> sample_type_list = list()\n",
      "> for (i in 1:nsamples) {\n",
      "+     samples_want = samples_data[samples_data[,1]==sample_types[i], 2]\n",
      "+     sample_type_list[[sample_types[i]]] = as.vector(samples_want)\n",
      "+ }\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> initial_matrix = data # store before doing various data transformations\n",
      "> data = log2(data+1)\n",
      "> sample_factoring = colnames(data)\n",
      "> for (i in 1:nsamples) {\n",
      "+     sample_type = sample_types[i]\n",
      "+     replicates_want = sample_type_list[[sample_type]]\n",
      "+     sample_factoring[ colnames(data) %in% replicates_want ] = sample_type\n",
      "+ }\n",
      "> sampleAnnotations = matrix(ncol=ncol(data),nrow=nsamples)\n",
      "> for (i in 1:nsamples) {\n",
      "+   sampleAnnotations[i,] = colnames(data) %in% sample_type_list[[sample_types[i]]]\n",
      "+ }\n",
      "> sampleAnnotations = apply(sampleAnnotations, 1:2, function(x) as.logical(x))\n",
      "> sampleAnnotations = sample_matrix_to_color_assignments(sampleAnnotations, col=sample_colors)\n",
      "> rownames(sampleAnnotations) = as.vector(sample_types)\n",
      "> colnames(sampleAnnotations) = colnames(data)\n",
      "> data = as.matrix(data) # convert to matrix\n",
      "> \n",
      "> # Centering rows\n",
      "> data = t(scale(t(data), scale=F))\n",
      "> \n",
      "> write.table(data, file=\"diffExpr.P0.05_C1.matrix.log2.centered.dat\", quote=F, sep='\t');\n",
      "> if (nrow(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two rows are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> if (ncol(data) < 2) { stop(\"\n",
      "+ \n",
      "+ **** Sorry, at least two columns are required for this matrix.\n",
      "+ \n",
      "+ \");}\n",
      "> sample_cor = cor(data, method='pearson', use='pairwise.complete.obs')\n",
      "> write.table(sample_cor, file=\"diffExpr.P0.05_C1.matrix.log2.centered.sample_cor.dat\", quote=F, sep='\t')\n",
      "> sample_dist = dist(t(data), method='euclidean')\n",
      "> hc_samples = hclust(sample_dist, method='complete')\n",
      "> pdf(\"diffExpr.P0.05_C1.matrix.log2.centered.sample_cor_matrix.pdf\")\n",
      "> sample_cor_for_plot = sample_cor\n",
      "> heatmap.3(sample_cor_for_plot, dendrogram='both', Rowv=as.dendrogram(hc_samples), Colv=as.dendrogram(hc_samples), col = myheatcol, scale='none', symm=TRUE, key=TRUE,density.info='none', trace='none', symkey=FALSE, symbreaks=F, margins=c(10,10), cexCol=1, cexRow=1, cex.main=0.75, main=paste(\"sample correlation matrix\n",
      "+ \", \"diffExpr.P0.05_C1.matrix.log2.centered\") , ColSideColors=sampleAnnotations, RowSideColors=t(sampleAnnotations))\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> gene_cor = NULL\n",
      "> gene_dist = dist(data, method='euclidean')\n",
      "> if (nrow(data) <= 1) { message('Too few genes to generate heatmap'); quit(status=0); }\n",
      "> hc_genes = hclust(gene_dist, method='complete')\n",
      "> heatmap_data = data\n",
      "> pdf(\"diffExpr.P0.05_C1.matrix.log2.centered.genes_vs_samples_heatmap.pdf\")\n",
      "> heatmap.3(heatmap_data, dendrogram='both', Rowv=as.dendrogram(hc_genes), Colv=as.dendrogram(hc_samples), col=myheatcol, scale=\"none\", density.info=\"none\", trace=\"none\", key=TRUE, keysize=1.2, cexCol=1, margins=c(10,10), cex.main=0.75, main=paste(\"samples vs. features\n",
      "+ \", \"diffExpr.P0.05_C1.matrix.log2.centered\" ) , ColSideColors=sampleAnnotations)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> save(list=ls(all=TRUE), file=\"diffExpr.P0.05_C1.matrix.RData\")\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "** Found 63 features as differentially expressed.\n",
      "\n",
      "CMD: /home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/PtR -m diffExpr.P0.05_C1.matrix --log2 --heatmap --min_colSums 0 --min_rowSums 0 --gene_dist euclidean --sample_dist euclidean --sample_cor_matrix --center_rows --save  -s /home/sam/gitrepos/coral_rnaseq_2017/scripts/montipora_b_vs_nb_44_vs_k4_trinity_sample_list.txt\n",
      "CMD: R --no-save --no-restore --no-site-file --no-init-file -q < diffExpr.P0.05_C1.matrix.R\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘fastcluster’\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    hclust\n",
      "\n",
      "for plotting:: min.raw: -0.500023824205951 max.raw: 1\n",
      "for plotting:: min.raw: -9.65319575785899 max.raw: 9.65319575785899\n",
      "\n",
      "real\t0m1.870s\n",
      "user\t0m1.796s\n",
      "sys\t0m0.052s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /media/sam/4TB_toshiba/montipora/20190129_sp_coral_transcripts/edgeR.22947.dir\n",
    "\n",
    "time \\\n",
    "/home/shared/Trinityrnaseq-v2.6.6/Analysis/DifferentialExpression/analyze_diff_expr.pl \\\n",
    "--matrix ${salmon_out_dir}/salmon.isoform.TMM.EXPR.matrix \\\n",
    "--samples ${samples} \\\n",
    "-C 1 \\\n",
    "-P 0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
